{
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "orig_nbformat": 2,
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.6.13 64-bit ('tissuespecific': conda)"
  },
  "interpreter": {
   "hash": "26aff48bbd2e903738c4a95289dc9f6861d7c456d77b24504d90c24e131deb55"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2,
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import os\n",
    "import requests\n",
    "from collections import defaultdict\n",
    "def insert_row(df, my_row):\n",
    "    df.loc[len(df)] = my_row"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "human_meta = pd.read_csv('metadata_ionbot_update_0401.csv', sep=';')\n",
    "human_meta = human_meta[human_meta['species'] == '[\"Homo sapiens (Human)\"]']\n",
    "human_meta = human_meta[['accession', 'url', 'title', 'projectDescription', 'species', 'tissues', 'ptms', 'instrumentNames', 'projectTags', 'experimentTypes', 'quantificationMethods', 'keywords','dataProcessingProtocol', 'submissionDate', 'references']]\n",
    "human_meta.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>url</th>\n",
       "      <th>title</th>\n",
       "      <th>projectDescription</th>\n",
       "      <th>species</th>\n",
       "      <th>tissues</th>\n",
       "      <th>ptms</th>\n",
       "      <th>instrumentNames</th>\n",
       "      <th>projectTags</th>\n",
       "      <th>experimentTypes</th>\n",
       "      <th>quantificationMethods</th>\n",
       "      <th>keywords</th>\n",
       "      <th>dataProcessingProtocol</th>\n",
       "      <th>submissionDate</th>\n",
       "      <th>references</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD003411</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Contribution of primary human fibroblasts and ...</td>\n",
       "      <td>While the most important players of inflammati...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"umbilical vein\"]</td>\n",
       "      <td>[{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...</td>\n",
       "      <td>[\"Q Exactive\"]</td>\n",
       "      <td>[\"Biological\",\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cell type-specific inflammatory response, hall...</td>\n",
       "      <td>Proteome Discoverer 1.4 (Thermo Fisher Scienti...</td>\n",
       "      <td>5/01/2016</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:27025457\"]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD001838</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Proteomics of EBV-infected lytic and refractor...</td>\n",
       "      <td>Most humans are infected with Epstein-Barr vir...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cell culture\"]</td>\n",
       "      <td>[{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...</td>\n",
       "      <td>[\"LTQ Orbitrap\"]</td>\n",
       "      <td>[\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>[\"Spectrum counting\"]</td>\n",
       "      <td>Epstein-Barr virus, lytic activation, proteomi...</td>\n",
       "      <td>Tandem mass spectra were extracted from raw fi...</td>\n",
       "      <td>27/02/2015</td>\n",
       "      <td>[[\"DOI:10.1128/JVI.00121-15\",\"PMID:25717101\"]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PXD009074</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Human T-cells EVs LC-MSMS -  Extracellular Ves...</td>\n",
       "      <td>As it is likely that different stimuli promote...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cell suspension culture\"]</td>\n",
       "      <td>[{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...</td>\n",
       "      <td>[\"LTQ Orbitrap\"]</td>\n",
       "      <td>[\"Biological\"]</td>\n",
       "      <td>[\"Gel-based experiment\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Apoptosis, EVs, microvesicles, exosomes, LC-MSMS</td>\n",
       "      <td>5. Two dimensional difference gel electrophore...</td>\n",
       "      <td>28/02/2018</td>\n",
       "      <td>[[\"DOI:10.3389/fimmu.2018.00534\",\"PMID:2959978...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PXD003943</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Host targets of Citrobacter secreted effectors</td>\n",
       "      <td>Identification of human proteins targeted by C...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cell culture\"]</td>\n",
       "      <td>[{\"ptm\":\"Oxidation\",\"UNIMOD\":\"35\"}]</td>\n",
       "      <td>[\"LTQ Orbitrap\"]</td>\n",
       "      <td>[\"Biological\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>effectors, host targets, type 3 secretion system</td>\n",
       "      <td>Data was searched with MSGF+ using PNNL's DMS ...</td>\n",
       "      <td>12/04/2016</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PXD000834</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Positional proteomics identification of the na...</td>\n",
       "      <td>Cytosolic carboxypeptidases (CCPs) constitute ...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"HEK-293FT cell\"]</td>\n",
       "      <td>[{\"ptm\":\"L-methionine (S)-sulfoxide\",\"UNIMOD\":...</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"PRIME-XS Project\",\"Biological\"]</td>\n",
       "      <td>[\"Bottom-up proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>degradomics C-terminomics carboxypeptidase</td>\n",
       "      <td>LC-MS/MS analysis of 2 ?L of the sample mixtur...</td>\n",
       "      <td>17/03/2014</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:25381060\"]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   accession                                                url  \\\n",
       "0  PXD003411  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "2  PXD001838  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "3  PXD009074  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "4  PXD003943  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "5  PXD000834  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "\n",
       "                                               title  \\\n",
       "0  Contribution of primary human fibroblasts and ...   \n",
       "2  Proteomics of EBV-infected lytic and refractor...   \n",
       "3  Human T-cells EVs LC-MSMS -  Extracellular Ves...   \n",
       "4     Host targets of Citrobacter secreted effectors   \n",
       "5  Positional proteomics identification of the na...   \n",
       "\n",
       "                                  projectDescription  \\\n",
       "0  While the most important players of inflammati...   \n",
       "2  Most humans are infected with Epstein-Barr vir...   \n",
       "3  As it is likely that different stimuli promote...   \n",
       "4  Identification of human proteins targeted by C...   \n",
       "5  Cytosolic carboxypeptidases (CCPs) constitute ...   \n",
       "\n",
       "                    species                      tissues  \\\n",
       "0  [\"Homo sapiens (Human)\"]           [\"umbilical vein\"]   \n",
       "2  [\"Homo sapiens (Human)\"]             [\"cell culture\"]   \n",
       "3  [\"Homo sapiens (Human)\"]  [\"cell suspension culture\"]   \n",
       "4  [\"Homo sapiens (Human)\"]             [\"cell culture\"]   \n",
       "5  [\"Homo sapiens (Human)\"]           [\"HEK-293FT cell\"]   \n",
       "\n",
       "                                                ptms         instrumentNames  \\\n",
       "0  [{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...          [\"Q Exactive\"]   \n",
       "2  [{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...        [\"LTQ Orbitrap\"]   \n",
       "3  [{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...        [\"LTQ Orbitrap\"]   \n",
       "4                [{\"ptm\":\"Oxidation\",\"UNIMOD\":\"35\"}]        [\"LTQ Orbitrap\"]   \n",
       "5  [{\"ptm\":\"L-methionine (S)-sulfoxide\",\"UNIMOD\":...  [\"LTQ Orbitrap Velos\"]   \n",
       "\n",
       "                         projectTags           experimentTypes  \\\n",
       "0        [\"Biological\",\"Biomedical\"]    [\"Shotgun proteomics\"]   \n",
       "2                     [\"Biomedical\"]    [\"Shotgun proteomics\"]   \n",
       "3                     [\"Biological\"]  [\"Gel-based experiment\"]   \n",
       "4                     [\"Biological\"]    [\"Shotgun proteomics\"]   \n",
       "5  [\"PRIME-XS Project\",\"Biological\"]  [\"Bottom-up proteomics\"]   \n",
       "\n",
       "   quantificationMethods                                           keywords  \\\n",
       "0                    NaN  Cell type-specific inflammatory response, hall...   \n",
       "2  [\"Spectrum counting\"]  Epstein-Barr virus, lytic activation, proteomi...   \n",
       "3                    NaN   Apoptosis, EVs, microvesicles, exosomes, LC-MSMS   \n",
       "4                    NaN   effectors, host targets, type 3 secretion system   \n",
       "5                    NaN         degradomics C-terminomics carboxypeptidase   \n",
       "\n",
       "                              dataProcessingProtocol submissionDate  \\\n",
       "0  Proteome Discoverer 1.4 (Thermo Fisher Scienti...      5/01/2016   \n",
       "2  Tandem mass spectra were extracted from raw fi...     27/02/2015   \n",
       "3  5. Two dimensional difference gel electrophore...     28/02/2018   \n",
       "4  Data was searched with MSGF+ using PNNL's DMS ...     12/04/2016   \n",
       "5  LC-MS/MS analysis of 2 ?L of the sample mixtur...     17/03/2014   \n",
       "\n",
       "                                          references  \n",
       "0                     [[\"DOI:null\",\"PMID:27025457\"]]  \n",
       "2     [[\"DOI:10.1128/JVI.00121-15\",\"PMID:25717101\"]]  \n",
       "3  [[\"DOI:10.3389/fimmu.2018.00534\",\"PMID:2959978...  \n",
       "4                                                NaN  \n",
       "5                     [[\"DOI:null\",\"PMID:25381060\"]]  "
      ]
     },
     "metadata": {},
     "execution_count": 2
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "print(human_meta.columns)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Index(['accession', 'url', 'title', 'projectDescription', 'species', 'tissues',\n",
      "       'ptms', 'instrumentNames', 'projectTags', 'experimentTypes',\n",
      "       'quantificationMethods', 'keywords', 'dataProcessingProtocol',\n",
      "       'submissionDate', 'references'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "source": [
    "row = ['PXD020192', 'https://www.ebi.ac.uk/pride/archive/projects/PXD020192', 'Proteomic Profiling of the Human Tissue and Biological Fluid Proteome', 'Here  we performed label-free liquid chromatography coupled to tandem mass spectrometry (LC-MS/MS) to profile the normal human proteome. In total  we analyzed 117 samples from 46 histologically normal tissues and organs at autopsy. Our high-resolution MS approach allowed for the quantification of 10 438 unique proteins. In order to expand our coverage of the human proteome  we combined our previously published biological fluid proteomic data from healthy individuals. We considered data from seven biological fluids  including urine  cerebrospinal fluid  synovial fluid  seminal plasma  sweat  cervical vaginal fluid  and nipple aspirate fluid. Overall  we generated tandem mass spectra corresponding to 13 028 unique human protein-coding genes.','[\"Homo sapiens (Human)\"]', '[\"Spleen\", \"Bone marrow\", \"Skin\", \"Lymph node\", \"Colon\", \"Liver\", \"Lung\", \"Kidney\", \"Brain\", \"Heart\", \"Placenta\", \"Testis\"]', '[{\"ptm\":\"iodoacetamide derivatized residue\",\"UNIMOD\":\"4\"}]', '[\"Q Exactive HF\"]', '[\"Biomedial\"]', '[\"Shotgun proteomics\"]', '', '', 'Data analysis XCalibur software v.2.0.6 (Thermo Fisher Scientific) was used to generate raw files. Raw files were uploaded into Proteome Discoverer v. 1.4 (Thermo Fisher Scientific) and searched with Sequest HT search engine against the Human 5640 Swiss-Prot protein database (January 2018). Search parameters included trypsin (full) enzyme digestion  a maximum of two missed cleavages  static modification of cysteine carbamidomethylation  dynamic modification of methionine oxidation  precursor mass tolerance of 7 ppm  fragment mass tolerance of 0.02 Da  1% false-discovery rate (FDR) at the peptide and protein level using the Percolator node.', '2020-04-07','[[\"DOI:10.1021/acs.jproteome.0c00502\", \"PMID:33107741\"]]']\n",
    "insert_row(human_meta, row)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "source": [
    "#find if all these pxds are in the conodes\n",
    "PXDs = human_meta[['accession']].values"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "source": [
    "reprocessed = 0\n",
    "reprocessed_projects = []\n",
    "for i in PXDs:\n",
    "    p = ''.join(list(i))\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.3/'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 53'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 54'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 55'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "\n",
    "print(reprocessed)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "PXD003411 was found in conode 53\n",
      "PXD004796 was found in conode 53\n",
      "PXD009914 was found in conode 53\n",
      "PXD000072 was found in conode 53\n",
      "PXD003409 was found in conode 53\n",
      "PXD003534 was found in conode 53\n",
      "PXD001674 was found in conode 53\n",
      "PXD005389 was found in conode 53\n",
      "PXD006833 was found in conode 53\n",
      "PXD001421 was found in conode 53\n",
      "PXD009747 was found in conode 53\n",
      "PXD006299 was found in conode 53\n",
      "PXD004626 was found in conode 53\n",
      "PXD007575 was found in conode 53\n",
      "PXD008581 was found in conode 53\n",
      "PXD009310 was found in conode 53\n",
      "PXD006578 was found in conode 53\n",
      "PXD002212 was found in conode 53\n",
      "PXD005554 was found in conode 53\n",
      "PXD001574 was found in conode 53\n",
      "PXD001609 was found in conode 53\n",
      "PXD007744 was found in conode 53\n",
      "PXD011467 was found in conode 53\n",
      "PXD010787 was found in conode 53\n",
      "PXD001357 was found in conode 53\n",
      "PXD004778 was found in conode 53\n",
      "PXD006847 was found in conode 53\n",
      "PXD006769 was found in conode 53\n",
      "PXD000685 was found in conode 53\n",
      "PXD009830 was found in conode 53\n",
      "PXD004764 was found in conode 53\n",
      "PXD009738 was found in conode 53\n",
      "PXD005508 was found in conode 53\n",
      "PXD008131 was found in conode 53\n",
      "PXD007726 was found in conode 53\n",
      "PXD010093 was found in conode 53\n",
      "PXD008939 was found in conode 53\n",
      "PXD000228 was found in conode 53\n",
      "PXD006421 was found in conode 53\n",
      "PXD002620 was found in conode 53\n",
      "PXD007592 was found in conode 53\n",
      "PXD002013 was found in conode 53\n",
      "PXD011124 was found in conode 53\n",
      "PXD005755 was found in conode 53\n",
      "PXD011204 was found in conode 53\n",
      "PXD006602 was found in conode 53\n",
      "PXD009302 was found in conode 53\n",
      "PXD004683 was found in conode 53\n",
      "PXD009750 was found in conode 53\n",
      "PXD003413 was found in conode 53\n",
      "PXD004788 was found in conode 53\n",
      "PXD009749 was found in conode 53\n",
      "PXD001423 was found in conode 53\n",
      "PXD000674 was found in conode 53\n",
      "PXD006201 was found in conode 53\n",
      "PXD002565 was found in conode 53\n",
      "PXD003045 was found in conode 53\n",
      "PXD007679 was found in conode 53\n",
      "PXD007935 was found in conode 53\n",
      "PXD005220 was found in conode 53\n",
      "PXD006522 was found in conode 53\n",
      "PXD003660 was found in conode 53\n",
      "PXD006618 was found in conode 53\n",
      "PXD009029 was found in conode 53\n",
      "PXD006268 was found in conode 53\n",
      "PXD009389 was found in conode 53\n",
      "PXD009996 was found in conode 53\n",
      "PXD005521 was found in conode 53\n",
      "PXD008260 was found in conode 53\n",
      "PXD000605 was found in conode 53\n",
      "PXD008940 was found in conode 53\n",
      "PXD006897 was found in conode 53\n",
      "PXD007270 was found in conode 53\n",
      "PXD011618 was found in conode 53\n",
      "PXD007906 was found in conode 53\n",
      "PXD009624 was found in conode 53\n",
      "PXD009751 was found in conode 53\n",
      "PXD010142 was found in conode 53\n",
      "PXD006114 was found in conode 53\n",
      "PXD004625 was found in conode 53\n",
      "PXD004059 was found in conode 53\n",
      "PXD009752 was found in conode 53\n",
      "PXD004624 was found in conode 53\n",
      "PXD005154 was found in conode 53\n",
      "PXD004682 was found in conode 53\n",
      "PXD006517 was found in conode 53\n",
      "PXD008713 was found in conode 53\n",
      "PXD009118 was found in conode 53\n",
      "PXD005739 was found in conode 53\n",
      "PXD008952 was found in conode 53\n",
      "PXD003971 was found in conode 53\n",
      "PXD006805 was found in conode 53\n",
      "PXD007153 was found in conode 53\n",
      "PXD001325 was found in conode 53\n",
      "PXD000533 was found in conode 53\n",
      "PXD003963 was found in conode 53\n",
      "PXD003410 was found in conode 53\n",
      "PXD009950 was found in conode 53\n",
      "PXD009041 was found in conode 53\n",
      "PXD003966 was found in conode 53\n",
      "PXD001249 was found in conode 53\n",
      "PXD009948 was found in conode 53\n",
      "PXD004510 was found in conode 53\n",
      "PXD001861 was found in conode 53\n",
      "PXD008715 was found in conode 53\n",
      "PXD001418 was found in conode 53\n",
      "PXD006576 was found in conode 53\n",
      "PXD004280 was found in conode 53\n",
      "PXD000445 was found in conode 53\n",
      "PXD008217 was found in conode 53\n",
      "PXD001359 was found in conode 53\n",
      "PXD003346 was found in conode 53\n",
      "PXD008311 was found in conode 53\n",
      "PXD008493 was found in conode 53\n",
      "PXD001326 was found in conode 53\n",
      "PXD000004 was found in conode 53\n",
      "PXD003415 was found in conode 53\n",
      "PXD001416 was found in conode 53\n",
      "PXD008272 was found in conode 53\n",
      "PXD007738 was found in conode 53\n",
      "PXD003215 was found in conode 53\n",
      "PXD005733 was found in conode 53\n",
      "PXD001068 was found in conode 53\n",
      "PXD002591 was found in conode 53\n",
      "PXD001422 was found in conode 53\n",
      "PXD003412 was found in conode 53\n",
      "PXD003965 was found in conode 53\n",
      "PXD008440 was found in conode 53\n",
      "PXD008150 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD003967 was found in conode 53\n",
      "PXD005477 was found in conode 53\n",
      "PXD009216 was found in conode 53\n",
      "PXD001524 was found in conode 53\n",
      "PXD009343 was found in conode 53\n",
      "PXD007846 was found in conode 53\n",
      "PXD006176 was found in conode 53\n",
      "PXD001417 was found in conode 53\n",
      "PXD009645 was found in conode 53\n",
      "PXD004154 was found in conode 53\n",
      "PXD003666 was found in conode 53\n",
      "PXD006465 was found in conode 53\n",
      "PXD006179 was found in conode 53\n",
      "PXD010786 was found in conode 53\n",
      "PXD008832 was found in conode 53\n",
      "PXD003408 was found in conode 53\n",
      "PXD005187 was found in conode 53\n",
      "PXD001324 was found in conode 53\n",
      "PXD002842 was found in conode 53\n",
      "PXD008723 was found in conode 53\n",
      "PXD011183 was found in conode 53\n",
      "PXD001118 was found in conode 53\n",
      "PXD007137 was found in conode 53\n",
      "PXD009266 was found in conode 53\n",
      "PXD007149 was found in conode 53\n",
      "PXD003594 was found in conode 53\n",
      "PXD008996 was found in conode 53\n",
      "PXD007890 was found in conode 53\n",
      "PXD006095 was found in conode 53\n",
      "PXD009295 was found in conode 53\n",
      "PXD009350 was found in conode 53\n",
      "PXD005190 was found in conode 53\n",
      "PXD006035 was found in conode 53\n",
      "PXD009366 was found in conode 53\n",
      "PXD008719 was found in conode 53\n",
      "PXD007624 was found in conode 53\n",
      "PXD006265 was found in conode 53\n",
      "PXD001898 was found in conode 53\n",
      "PXD006590 was found in conode 53\n",
      "PXD003414 was found in conode 53\n",
      "PXD003709 was found in conode 53\n",
      "PXD008029 was found in conode 53\n",
      "PXD009737 was found in conode 53\n",
      "PXD006776 was found in conode 53\n",
      "PXD006591 was found in conode 53\n",
      "PXD008205 was found in conode 53\n",
      "PXD002436 was found in conode 53\n",
      "PXD001985 was found in conode 53\n",
      "PXD003132 was found in conode 53\n",
      "PXD002211 was found in conode 53\n",
      "PXD007745 was found in conode 53\n",
      "PXD006570 was found in conode 53\n",
      "PXD001017 was found in conode 53\n",
      "PXD002990 was found in conode 53\n",
      "PXD006096 was found in conode 53\n",
      "PXD001415 was found in conode 53\n",
      "PXD006589 was found in conode 53\n",
      "PXD008158 was found in conode 53\n",
      "PXD010690 was found in conode 53\n",
      "PXD003075 was found in conode 53\n",
      "PXD009021 was found in conode 53\n",
      "PXD003226 was found in conode 53\n",
      "PXD010150 was found in conode 53\n",
      "PXD006571 was found in conode 53\n",
      "PXD003406 was found in conode 53\n",
      "PXD001420 was found in conode 53\n",
      "PXD010708 was found in conode 53\n",
      "PXD001541 was found in conode 53\n",
      "PXD003657 was found in conode 53\n",
      "PXD004355 was found in conode 53\n",
      "PXD005055 was found in conode 53\n",
      "PXD009142 was found in conode 53\n",
      "PXD009895 was found in conode 53\n",
      "PXD009803 was found in conode 53\n",
      "PXD004540 was found in conode 53\n",
      "PXD008933 was found in conode 53\n",
      "PXD009801 was found in conode 53\n",
      "PXD006561 was found in conode 53\n",
      "PXD005590 was found in conode 53\n",
      "PXD007271 was found in conode 53\n",
      "PXD010073 was found in conode 53\n",
      "PXD001677 was found in conode 53\n",
      "PXD007258 was found in conode 53\n",
      "PXD003417 was found in conode 53\n",
      "PXD008465 was found in conode 53\n",
      "PXD003407 was found in conode 53\n",
      "PXD009437 was found in conode 53\n",
      "PXD002883 was found in conode 53\n",
      "PXD006476 was found in conode 53\n",
      "PXD007750 was found in conode 53\n",
      "PXD001419 was found in conode 53\n",
      "PXD009754 was found in conode 53\n",
      "PXD008795 was found in conode 53\n",
      "PXD008035 was found in conode 53\n",
      "PXD001327 was found in conode 53\n",
      "PXD004415 was found in conode 53\n",
      "PXD003053 was found in conode 53\n",
      "PXD003416 was found in conode 53\n",
      "PXD006572 was found in conode 53\n",
      "PXD008934 was found in conode 53\n",
      "PXD009755 was found in conode 53\n",
      "PXD007262 was found in conode 53\n",
      "PXD006552 was found in conode 53\n",
      "PXD009646 was found in conode 53\n",
      "PXD009084 was found in conode 53\n",
      "PXD005981 was found in conode 53\n",
      "PXD002354 was found in conode 53\n",
      "PXD002057 was found in conode 53\n",
      "PXD010698 was found in conode 53\n",
      "PXD010793 was found in conode 54\n",
      "PXD008381 was found in conode 54\n",
      "PXD009797 was found in conode 54\n",
      "PXD004086 was found in conode 54\n",
      "PXD002960 was found in conode 53\n",
      "PXD009181 was found in conode 54\n",
      "PXD004320 was found in conode 54\n",
      "PXD002132 was found in conode 54\n",
      "PXD010424 was found in conode 54\n",
      "PXD006386 was found in conode 54\n",
      "PXD008033 was found in conode 54\n",
      "PXD010103 was found in conode 54\n",
      "PXD010520 was found in conode 54\n",
      "PXD009868 was found in conode 54\n",
      "PXD003294 was found in conode 54\n",
      "PXD010294 was found in conode 54\n",
      "PXD004334 was found in conode 54\n",
      "PXD008682 was found in conode 54\n",
      "PXD007898 was found in conode 54\n",
      "PXD009870 was found in conode 54\n",
      "PXD008178 was found in conode 54\n",
      "PXD006863 was found in conode 54\n",
      "PXD003296 was found in conode 54\n",
      "PXD009149 was found in conode 54\n",
      "PXD005650 was found in conode 54\n",
      "PXD003297 was found in conode 54\n",
      "PXD009396 was found in conode 54\n",
      "PXD002437 was found in conode 54\n",
      "PXD005045 was found in conode 54\n",
      "PXD007234 was found in conode 54\n",
      "PXD009234 was found in conode 54\n",
      "PXD010223 was found in conode 54\n",
      "PXD009438 was found in conode 54\n",
      "PXD008870 was found in conode 54\n",
      "PXD008925 was found in conode 54\n",
      "PXD009112 was found in conode 54\n",
      "PXD007123 was found in conode 54\n",
      "PXD010630 was found in conode 54\n",
      "PXD009487 was found in conode 54\n",
      "PXD000529 was found in conode 54\n",
      "PXD009196 was found in conode 54\n",
      "PXD009724 was found in conode 54\n",
      "PXD008467 was found in conode 54\n",
      "PXD003293 was found in conode 54\n",
      "PXD010634 was found in conode 54\n",
      "PXD010132 was found in conode 54\n",
      "PXD006672 was found in conode 54\n",
      "PXD009360 was found in conode 54\n",
      "PXD011881 was found in conode 54\n",
      "PXD011309 was found in conode 54\n",
      "PXD005914 was found in conode 54\n",
      "PXD009933 was found in conode 54\n",
      "PXD009161 was found in conode 54\n",
      "PXD009652 was found in conode 53\n",
      "PXD008252 was found in conode 54\n",
      "PXD008454 was found in conode 54\n",
      "PXD002672 was found in conode 54\n",
      "PXD002014 was found in conode 54\n",
      "PXD005154 was found in conode 53\n",
      "PXD000443 was found in conode 54\n",
      "PXD000440 was found in conode 54\n",
      "PXD008351 was found in conode 54\n",
      "PXD002097 was found in conode 54\n",
      "PXD007761 was found in conode 54\n",
      "PXD006856 was found in conode 54\n",
      "PXD004001 was found in conode 54\n",
      "PXD010538 was found in conode 54\n",
      "PXD006738 was found in conode 54\n",
      "PXD009936 was found in conode 54\n",
      "PXD010005 was found in conode 54\n",
      "PXD006581 was found in conode 54\n",
      "PXD009709 was found in conode 54\n",
      "PXD008443 was found in conode 54\n",
      "PXD010284 was found in conode 54\n",
      "PXD008601 was found in conode 54\n",
      "PXD010260 was found in conode 54\n",
      "PXD007080 was found in conode 54\n",
      "PXD005323 was found in conode 54\n",
      "PXD009144 was found in conode 54\n",
      "PXD009781 was found in conode 54\n",
      "PXD006178 was found in conode 54\n",
      "PXD003544 was found in conode 54\n",
      "PXD011301 was found in conode 54\n",
      "PXD011152 was found in conode 54\n",
      "PXD008147 was found in conode 54\n",
      "PXD010222 was found in conode 54\n",
      "PXD008463 was found in conode 54\n",
      "PXD009440 was found in conode 54\n",
      "PXD006911 was found in conode 54\n",
      "PXD009254 was found in conode 54\n",
      "PXD009056 was found in conode 54\n",
      "PXD008338 was found in conode 54\n",
      "PXD009600 was found in conode 54\n",
      "PXD007671 was found in conode 54\n",
      "PXD010805 was found in conode 54\n",
      "PXD006335 was found in conode 54\n",
      "PXD007122 was found in conode 54\n",
      "PXD010941 was found in conode 54\n",
      "PXD009015 was found in conode 54\n",
      "PXD011241 was found in conode 54\n",
      "PXD002687 was found in conode 54\n",
      "PXD009931 was found in conode 54\n",
      "PXD010942 was found in conode 54\n",
      "PXD007921 was found in conode 54\n",
      "PXD010527 was found in conode 54\n",
      "PXD009477 was found in conode 54\n",
      "PXD007705 was found in conode 54\n",
      "PXD009138 was found in conode 54\n",
      "PXD011033 was found in conode 54\n",
      "PXD007894 was found in conode 54\n",
      "PXD008497 was found in conode 54\n",
      "PXD002133 was found in conode 54\n",
      "PXD007959 was found in conode 54\n",
      "PXD005465 was found in conode 54\n",
      "PXD009741 was found in conode 54\n",
      "PXD009158 was found in conode 54\n",
      "PXD008518 was found in conode 54\n",
      "PXD011256 was found in conode 54\n",
      "PXD009442 was found in conode 54\n",
      "PXD010393 was found in conode 54\n",
      "PXD009258 was found in conode 54\n",
      "PXD009241 was found in conode 54\n",
      "PXD010286 was found in conode 54\n",
      "PXD009476 was found in conode 54\n",
      "PXD006697 was found in conode 54\n",
      "PXD009206 was found in conode 54\n",
      "PXD009606 was found in conode 54\n",
      "PXD010124 was found in conode 54\n",
      "PXD011960 was found in conode 54\n",
      "PXD011009 was found in conode 54\n",
      "PXD010821 was found in conode 54\n",
      "PXD008918 was found in conode 54\n",
      "PXD009760 was found in conode 54\n",
      "PXD003743 was found in conode 54\n",
      "PXD010406 was found in conode 54\n",
      "PXD009327 was found in conode 54\n",
      "PXD007662 was found in conode 54\n",
      "PXD009916 was found in conode 54\n",
      "PXD009092 was found in conode 54\n",
      "PXD009444 was found in conode 54\n",
      "PXD002825 was found in conode 54\n",
      "PXD009267 was found in conode 54\n",
      "PXD003252 was found in conode 54\n",
      "PXD006497 was found in conode 54\n",
      "PXD006371 was found in conode 54\n",
      "PXD001873 was found in conode 54\n",
      "PXD009025 was found in conode 54\n",
      "PXD009276 was found in conode 54\n",
      "PXD003086 was found in conode 54\n",
      "PXD007896 was found in conode 54\n",
      "PXD009900 was found in conode 54\n",
      "PXD008115 was found in conode 54\n",
      "PXD010122 was found in conode 54\n",
      "PXD005544 was found in conode 54\n",
      "PXD009380 was found in conode 54\n",
      "PXD002598 was found in conode 54\n",
      "PXD005945 was found in conode 54\n",
      "PXD010372 was found in conode 54\n",
      "PXD007762 was found in conode 54\n",
      "PXD009893 was found in conode 54\n",
      "PXD010027 was found in conode 54\n",
      "PXD009839 was found in conode 54\n",
      "PXD005436 was found in conode 54\n",
      "PXD011492 was found in conode 54\n",
      "PXD010075 was found in conode 54\n",
      "PXD010202 was found in conode 54\n",
      "PXD009915 was found in conode 54\n",
      "PXD009596 was found in conode 54\n",
      "PXD001460 was found in conode 54\n",
      "PXD009784 was found in conode 54\n",
      "PXD003533 was found in conode 54\n",
      "PXD008130 was found in conode 54\n",
      "PXD009592 was found in conode 54\n",
      "PXD008197 was found in conode 54\n",
      "PXD009035 was found in conode 54\n",
      "PXD009374 was found in conode 54\n",
      "PXD009880 was found in conode 54\n",
      "PXD006607 was found in conode 54\n",
      "PXD011242 was found in conode 54\n",
      "PXD003587 was found in conode 54\n",
      "PXD010705 was found in conode 54\n",
      "PXD010155 was found in conode 54\n",
      "PXD009095 was found in conode 54\n",
      "PXD011000 was found in conode 54\n",
      "PXD009603 was found in conode 54\n",
      "PXD009819 was found in conode 54\n",
      "PXD010061 was found in conode 54\n",
      "PXD009838 was found in conode 54\n",
      "PXD007543 was found in conode 54\n",
      "PXD009367 was found in conode 54\n",
      "PXD008863 was found in conode 54\n",
      "PXD006788 was found in conode 54\n",
      "PXD007263 was found in conode 54\n",
      "PXD011175 was found in conode 54\n",
      "PXD009110 was found in conode 54\n",
      "PXD008693 was found in conode 54\n",
      "PXD007675 was found in conode 54\n",
      "PXD003362 was found in conode 54\n",
      "PXD010870 was found in conode 54\n",
      "PXD007740 was found in conode 54\n",
      "PXD006723 was found in conode 54\n",
      "PXD009759 was found in conode 54\n",
      "PXD010210 was found in conode 54\n",
      "PXD001425 was found in conode 54\n",
      "PXD009317 was found in conode 54\n",
      "PXD008584 was found in conode 54\n",
      "PXD009935 was found in conode 54\n",
      "PXD007081 was found in conode 54\n",
      "PXD008468 was found in conode 54\n",
      "PXD011456 was found in conode 54\n",
      "PXD008127 was found in conode 54\n",
      "PXD005709 was found in conode 54\n",
      "PXD004443 was found in conode 54\n",
      "PXD010469 was found in conode 54\n",
      "PXD006743 was found in conode 54\n",
      "PXD010510 was found in conode 54\n",
      "PXD008012 was found in conode 54\n",
      "PXD009453 was found in conode 54\n",
      "PXD011609 was found in conode 54\n",
      "PXD008896 was found in conode 54\n",
      "PXD003323 was found in conode 54\n",
      "PXD003120 was found in conode 54\n",
      "PXD008525 was found in conode 54\n",
      "PXD008354 was found in conode 54\n",
      "PXD011146 was found in conode 54\n",
      "PXD009300 was found in conode 54\n",
      "PXD000426 was found in conode 54\n",
      "PXD010776 was found in conode 54\n",
      "PXD006404 was found in conode 54\n",
      "PXD009716 was found in conode 54\n",
      "PXD007873 was found in conode 54\n",
      "PXD005693 was found in conode 54\n",
      "PXD007049 was found in conode 54\n",
      "PXD009570 was found in conode 54\n",
      "PXD007231 was found in conode 54\n",
      "PXD008992 was found in conode 54\n",
      "PXD006419 was found in conode 54\n",
      "PXD003295 was found in conode 54\n",
      "PXD007515 was found in conode 54\n",
      "PXD007594 was found in conode 54\n",
      "PXD009508 was found in conode 54\n",
      "PXD009197 was found in conode 54\n",
      "PXD010466 was found in conode 54\n",
      "PXD009174 was found in conode 54\n",
      "PXD001682 was found in conode 54\n",
      "PXD010088 was found in conode 54\n",
      "PXD009393 was found in conode 54\n",
      "PXD009565 was found in conode 54\n",
      "PXD009677 was found in conode 54\n",
      "PXD007990 was found in conode 54\n",
      "PXD008568 was found in conode 54\n",
      "PXD007759 was found in conode 54\n",
      "PXD008005 was found in conode 54\n",
      "PXD005444 was found in conode 54\n",
      "PXD001862 was found in conode 54\n",
      "PXD009553 was found in conode 54\n",
      "PXD006269 was found in conode 54\n",
      "PXD010777 was found in conode 54\n",
      "PXD009891 was found in conode 54\n",
      "PXD009348 was found in conode 54\n",
      "PXD008562 was found in conode 54\n",
      "PXD005944 was found in conode 54\n",
      "PXD008824 was found in conode 54\n",
      "PXD009917 was found in conode 54\n",
      "PXD008978 was found in conode 54\n",
      "PXD009500 was found in conode 54\n",
      "PXD005301 was found in conode 54\n",
      "PXD007975 was found in conode 54\n",
      "PXD011214 was found in conode 54\n",
      "PXD009813 was found in conode 54\n",
      "PXD006215 was found in conode 54\n",
      "PXD008333 was found in conode 54\n",
      "PXD008575 was found in conode 54\n",
      "PXD009292 was found in conode 54\n",
      "PXD009061 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 55\n",
      "PXD014300 was found in conode 55\n",
      "PXD008299 was found in conode 55\n",
      "PXD015377 was found in conode 55\n",
      "PXD015523 was found in conode 55\n",
      "PXD018894 was found in conode 55\n",
      "PXD018450 was found in conode 55\n",
      "PXD018625 was found in conode 55\n",
      "PXD018621 was found in conode 55\n",
      "PXD017213 was found in conode 55\n",
      "PXD018226 was found in conode 55\n",
      "PXD015653 was found in conode 55\n",
      "PXD004928 was found in conode 55\n",
      "PXD012971 was found in conode 55\n",
      "PXD012970 was found in conode 55\n",
      "PXD017541 was found in conode 55\n",
      "PXD017391 was found in conode 55\n",
      "PXD016060 was found in conode 55\n",
      "PXD016836 was found in conode 55\n",
      "PXD018270 was found in conode 55\n",
      "PXD017531 was found in conode 55\n",
      "PXD015441 was found in conode 55\n",
      "PXD013219 was found in conode 55\n",
      "PXD018502 was found in conode 55\n",
      "PXD016618 was found in conode 55\n",
      "PXD011178 was found in conode 55\n",
      "PXD016994 was found in conode 55\n",
      "PXD015656 was found in conode 55\n",
      "PXD012960 was found in conode 55\n",
      "PXD018918 was found in conode 55\n",
      "PXD018880 was found in conode 55\n",
      "PXD018588 was found in conode 55\n",
      "PXD019251 was found in conode 55\n",
      "PXD017586 was found in conode 55\n",
      "PXD019204 was found in conode 55\n",
      "PXD017468 was found in conode 55\n",
      "PXD017427 was found in conode 55\n",
      "PXD014254 was found in conode 55\n",
      "PXD015039 was found in conode 55\n",
      "PXD016126 was found in conode 55\n",
      "PXD017261 was found in conode 55\n",
      "PXD016865 was found in conode 55\n",
      "PXD016968 was found in conode 55\n",
      "PXD015863 was found in conode 55\n",
      "PXD017109 was found in conode 55\n",
      "PXD017642 was found in conode 55\n",
      "PXD018449 was found in conode 55\n",
      "PXD010508 was found in conode 55\n",
      "PXD011961 was found in conode 55\n",
      "PXD013541 was found in conode 55\n",
      "PXD016261 was found in conode 55\n",
      "PXD005970 was found in conode 55\n",
      "PXD018143 was found in conode 55\n",
      "PXD018294 was found in conode 55\n",
      "PXD017268 was found in conode 55\n",
      "PXD016882 was found in conode 55\n",
      "PXD018678 was found in conode 55\n",
      "PXD017976 was found in conode 55\n",
      "PXD018330 was found in conode 55\n",
      "PXD015453 was found in conode 55\n",
      "PXD018661 was found in conode 55\n",
      "PXD005514 was found in conode 55\n",
      "PXD016437 was found in conode 55\n",
      "PXD017717 was found in conode 55\n",
      "PXD015679 was found in conode 55\n",
      "PXD016555 was found in conode 55\n",
      "PXD015836 was found in conode 55\n",
      "PXD018061 was found in conode 55\n",
      "PXD017804 was found in conode 55\n",
      "PXD015817 was found in conode 55\n",
      "PXD018682 was found in conode 55\n",
      "PXD015847 was found in conode 55\n",
      "PXD018321 was found in conode 55\n",
      "PXD013517 was found in conode 55\n",
      "PXD016832 was found in conode 55\n",
      "PXD017011 was found in conode 55\n",
      "PXD018035 was found in conode 55\n",
      "PXD018078 was found in conode 55\n",
      "PXD015780 was found in conode 55\n",
      "PXD018186 was found in conode 55\n",
      "PXD017656 was found in conode 55\n",
      "PXD017792 was found in conode 55\n",
      "PXD017983 was found in conode 55\n",
      "PXD017030 was found in conode 55\n",
      "PXD017023 was found in conode 55\n",
      "PXD018048 was found in conode 55\n",
      "PXD017989 was found in conode 55\n",
      "PXD016151 was found in conode 55\n",
      "PXD015877 was found in conode 55\n",
      "PXD012422 was found in conode 55\n",
      "PXD005978 was found in conode 55\n",
      "PXD016678 was found in conode 55\n",
      "PXD016782 was found in conode 55\n",
      "PXD013572 was found in conode 55\n",
      "PXD012615 was found in conode 55\n",
      "PXD017056 was found in conode 55\n",
      "PXD014894 was found in conode 55\n",
      "PXD015163 was found in conode 55\n",
      "PXD013616 was found in conode 55\n",
      "PXD018208 was found in conode 55\n",
      "PXD014381 was found in conode 55\n",
      "PXD013649 was found in conode 55\n",
      "PXD015239 was found in conode 55\n",
      "PXD014199 was found in conode 55\n",
      "PXD005611 was found in conode 55\n",
      "PXD012732 was found in conode 55\n",
      "PXD013038 was found in conode 55\n",
      "PXD013075 was found in conode 55\n",
      "PXD011116 was found in conode 55\n",
      "PXD011129 was found in conode 55\n",
      "PXD015518 was found in conode 55\n",
      "PXD015122 was found in conode 55\n",
      "PXD012029 was found in conode 55\n",
      "PXD014369 was found in conode 55\n",
      "PXD014329 was found in conode 55\n",
      "PXD005093 was found in conode 55\n",
      "PXD013064 was found in conode 55\n",
      "PXD014524 was found in conode 55\n",
      "PXD015571 was found in conode 55\n",
      "PXD017482 was found in conode 55\n",
      "PXD009261 was found in conode 55\n",
      "PXD016999 was found in conode 55\n",
      "PXD020192 was found in conode 55\n",
      "638\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "print(len(reprocessed_projects))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "638\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "source": [
    "x = list(set(PXDs.flatten()))\n",
    "y = list(x)\n",
    "not_repr = len(list(set(y) - set(reprocessed_projects)))\n",
    "total = len(y)\n",
    "print('{} projects from the total of {} were not found in the reprocessed data'.format(not_repr, total))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "708 projects from the total of 1344 were not found in the reprocessed data\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Filter the metadata file with only the reprocessed projects with a data folder + clean up to find only tissues and not cell cultures"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "source": [
    "human_meta_reprocessed = human_meta[human_meta['accession'].isin(reprocessed_projects)]\n",
    "print(human_meta.shape)\n",
    "print(human_meta_reprocessed.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(1346, 15)\n",
      "(637, 15)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "source": [
    "human_meta_reprocessed['tissues'].unique()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "array(['[\"umbilical vein\"]', '[\"SW-480 cell\",\"colon\"]', nan,\n",
       "       '[\"blood platelet\"]', '[\"blood\"]', '[\"HCT-116 cell\",\"colon\"]',\n",
       "       '[\"liver\"]', '[\"HEp-2 cell\",\"JURKAT cell\"]', '[\"blood serum\"]',\n",
       "       '[\"cell culture\"]',\n",
       "       '[\"epithelial ovarian cancer cell\",\"mammary gland\"]',\n",
       "       '[\"HeLa cell\",\"HEK-293 cell\"]', '[\"lung\"]', '[\"dental plaque\"]',\n",
       "       '[\"pancreatic islet\"]', '[\"breast\"]', '[\"testis\"]', '[\"ureter\"]',\n",
       "       '[\"skin\"]', '[\"blood plasma\"]', '[\"HeLa cell\"]',\n",
       "       '[\"blood\",\"larynx squamous cell carcinoma cell line\"]',\n",
       "       '[\"melanoma cell\"]', '[\"urine\"]', '[\"cell culture\",\"spleen\"]',\n",
       "       '[\"synovial tissue\"]', '[\"kidney\"]', '[\"embryo\"]',\n",
       "       '[\"cerebrospinal fluid\",\"brain\",\"blood\"]',\n",
       "       '[\"kidney\",\"blood plasma\",\"cell culture\",\"HeLa cell\"]',\n",
       "       '[\"trophoblast\"]', '[\"urinary bladder\"]',\n",
       "       '[\"cerebral gray matter\"]', '[\"JURKAT cell\"]', '[\"colon\"]',\n",
       "       '[\"cell culture\",\"HeLa cell\"]', '[\"brain\"]',\n",
       "       '[\"palatine tonsil\",\"parotid gland\",\"nasal polyp\"]', '[\"cranium\"]',\n",
       "       '[\"temporal lobe\",\"substantia nigra\",\"ovary\",\"blood plasma\",\"liver\",\"pancreas\",\"blood serum\",\"cerebrospinal fluid\",\"monocyte\"]',\n",
       "       '[\"blood plasma\",\"blood serum\"]', '[\"ovary\"]',\n",
       "       '[\"uterine cervix\"]', '[\"pancreatic cancer cell\"]',\n",
       "       '[\"anterior chamber of the eye\"]', '[\"cell suspension culture\"]',\n",
       "       '[\"blood plasma\",\"cerebrospinal fluid\"]', '[\"heart\"]',\n",
       "       '[\"sputum\"]', '[\"lymph node\"]', '[\"cerebrospinal fluid\"]',\n",
       "       '[\"pancreas\"]',\n",
       "       '[\"cerebral cortex\",\"arcuate nucleus\",\"hypothalamic nucleus\",\"paraventricular nucleus of hypothalamus\"]',\n",
       "       '[\"breast cancer cell line\"]',\n",
       "       '[\"head and neck squamous cell carcinoma cell line\"]',\n",
       "       '[\"prostate adenocarcinoma cell\"]',\n",
       "       '[\"peritoneal dialysis fluid\"]', '[\"bronchoalveolar lavage\"]',\n",
       "       '[\"feces\"]', '[\"HeLa cell\",\"placenta\"]', '[\"skeletal muscle\"]',\n",
       "       '[\"knee\",\"chondrocyte\"]', '[\"head\",\"neck\"]', '[\"tooth enamel\"]',\n",
       "       '[\"cell culture\",\"colon\"]', '[\"saliva\"]',\n",
       "       '[\"cell culture\",\"reticulocyte\",\"erythrocyte\"]',\n",
       "       '[\"prostate cancer cell\"]',\n",
       "       '[\"kidney\",\"cell culture\",\"blood serum\"]', '[\"retina\"]',\n",
       "       '[\"non-small cell lung cancer cell\"]',\n",
       "       '[\"lung\",\"HeLa cell\",\"brain\"]',\n",
       "       '[\"JURKAT cell\",\"cell culture\",\"HeLa cell\",\"cell suspension culture\"]',\n",
       "       '[\"epithelial ovarian cancer cell\"]',\n",
       "       '[\"adipose tissue\",\"umbilical cord\"]', '[\"follicular fluid\"]',\n",
       "       '[\"cervical epithelial cell\"]',\n",
       "       '[\"skin\",\"cell suspension culture\"]', '[\"cell culture\",\"breast\"]',\n",
       "       '[\"vitreous humor\"]', '[\"breast cancer cell line\",\"cell culture\"]',\n",
       "       '[\"esophagus\"]', '[\"small intestine\"]',\n",
       "       '[\"lung\",\"cell culture\",\"bone marrow\"]', '[\"microvessel\"]',\n",
       "       '[\"ARPE-19 cell\"]', '[\"oocyte\"]',\n",
       "       '[\"prostate gland\",\"gut\",\"testis\",\"helper T-lymphocyte\",\"kidney\",\"retina\",\"pancreas\",\"adrenal gland\",\"spinal cord\",\"natural killer cell\",\"cytotoxic T-lymphocyte\",\"liver\",\"urinary bladder\",\"gall bladder\",\"brain\",\"heart\",\"lung\",\"ovary\",\"blood platelet\",\"frontal lobe\",\"B-lymphocyte\",\"colon\",\"placenta\",\"monocyte\",\"rectum\",\"esophagus\"]',\n",
       "       '[\"cervicovaginal fluid\",\"CACO-2 cell\",\"cardiomyocyte cell line\",\"blood plasma\",\"cell culture\",\"HeLa cell\",\"adipocyte\",\"placenta\",\"brain\",\"HepG2 cell\"]',\n",
       "       '[\"prostate gland\"]',\n",
       "       '[\"activated skeletal muscle satellite cell\"]', '[\"THP-1 cell\"]',\n",
       "       '[\"primary cell\"]', '[\"stomach\"]', '[\"HEK-293T cell\"]',\n",
       "       '[\"ascites\",\"macrophage\"]', '[\"skin fibroblast\"]',\n",
       "       '[\"medulloblastoma cell\"]', '[\"kidney\",\"embryo\"]',\n",
       "       '[\"whole body\"]', '[\"kidney\",\"blood plasma\",\"blood\"]',\n",
       "       '[\"cornea\"]',\n",
       "       '[\"liver\",\"spleen\",\"skin\",\"thyroid gland\",\"stomach\",\"artery\",\"testis\",\"brain\",\"heart\",\"saliva\",\"lung\",\"ovary\",\"colon\",\"small intestine\",\"skeletal muscle\",\"esophagus\"]',\n",
       "       '[\"Spleen\", \"Bone marrow\", \"Skin\", \"Lymph node\", \"Colon\", \"Liver\", \"Lung\", \"Kidney\", \"Brain\", \"Heart\", \"Placenta\", \"Testis\"]'],\n",
       "      dtype=object)"
      ]
     },
     "metadata": {},
     "execution_count": 10
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    "#manually selected the not cell culture tissues\n",
    "keep_list = ['[\"umbilical vein\"]','[\"eye\"]','[\"blood platelet\"]','[\"urinary bladder\"]','[\"blood\"]','[\"liver\"]','[\"skeletal muscle\"]','[\"lung\"]','[\"blood plasma\"]','[\"blood serum\"]','[\"oral mucosa\"]','[\"brain\"]','[\"kidney\"]','[\"urine\"]','[\"dental plaque\"]','[\"articular cartilage\",\"hip joint\"]','[\"pancreatic islet\"]','[\"saliva\"]','[\"liver\",\"brain\"]','[\"breast\"]','[\"testis\"]','[\"bile\"]','[\"ureter\"]','[\"skin\"]','[\"cerebrospinal fluid\"]','[\"synovial tissue\"]','[\"ascites\"]','[\"blood plasma\",\"blood serum\"]','[\"cornea\"]','[\"ascites\",\"blood serum\"]','[\"lung\",\"pleura\"]','[\"cerebrospinal fluid\",\"brain\",\"blood\"]','[\"cervicovaginal fluid\"]','[\"B-lymphocyte\"]','[\"retinal pigment epithelium cell\"]','[\"kidney\",\"blood plasma\",\"cell culture\",\"HeLa cell\"]','[\"lung\",\"liver\"]','[\"colon\"]','[\"spleen\"]','[\"trophoblast\"]','[\"carotid artery\"]','[\"endometrial gland\"]','[\"urinary bladder\",\"colon\"]','[\"heart\"]','[\"cerebral gray matter\"]','[\"kidney\",\"urine\"]','[\"pancreatic ductal adenocarcinoma cell\",\"cholangiocarcinoma cell\"]','[\"primary cell\"]','[\"palatine tonsil\",\"parotid gland\",\"nasal polyp\"]','[\"bone marrow\"]','[\"thyroid gland\",\"brain\"]','[\"duodenum\"]','[\"cranium\"]','[\"temporal lobe\",\"substantia nigra\",\"ovary\",\"blood plasma\",\"liver\",\"pancreas\",\"blood serum\",\"cerebrospinal fluid\",\"monocyte\"]','[\"blood vessel\"]','[\"prostate gland\"]','[\"ovary\"]','[\"uterine cervix\"]','[\"primary cell\",\"testis\"]','[\"anterior chamber of the eye\"]','[\"blood plasma\",\"cerebrospinal fluid\"]','[\"pancreatic islet beta cell\"]','[\"sputum\"]','[\"lymph node\"]','[\"semen\",\"spermatozoid\"]','[\"pancreas\"]','[\"eyelid\",\"conjunctiva\",\"orbit\"]','[\"lens cortex\"]','[\"tendon\"]','[\"sweat\",\"skin\"]','[\"heart valve\"]','[\"placenta\"]','[\"lung\",\"thorax\"]','[\"skin\",\"ciliary body\"]','[\"cerebral cortex\",\"arcuate nucleus\",\"hypothalamic nucleus\",\"paraventricular nucleus of hypothalamus\"]','[\"peritoneal cavity\"]','[\"oocyte\"]','[\"peritoneal dialysis fluid\"]','[\"brain\",\"bone marrow\"]','[\"fibroblast\"]','[\"bronchoalveolar lavage\"]','[\"feces\"]','[\"knee\",\"chondrocyte\"]','[\"head\",\"neck\"]','[\"tooth enamel\"]','[\"retina\"]','[\"hair follicle dermal papilla cell\"]','[\"peripheral blood mononuclear cell\"]','[\"erythrocyte\"]','[\"tongue\"]','[\"adipose tissue\",\"umbilical cord\"]','[\"follicular fluid\"]','[\"blood platelet\",\"blood plasma\",\"blood serum\",\"blood\",\"erythrocyte\"]','[\"vitreous humor\"]','[\"saliva\",\"blood plasma\",\"colostrum\"]','[\"vermiform appendix\"]','[\"respiratory mucus\"]','[\"liver\",\"colon\"]','[\"esophagus\"]','[\"small intestine\"]','[\"microvessel\"]','[\"mesenchymal stem cell\",\"haematopoietic stem cells\",\"bone marrow stromal stem cell\"]','[\"kidney\",\"brain\"]','[\"prostate gland\",\"gut\",\"testis\",\"helper T-lymphocyte\",\"kidney\",\"retina\",\"pancreas\",\"adrenal gland\",\"spinal cord\",\"natural killer cell\",\"cytotoxic T-lymphocyte\",\"liver\",\"urinary bladder\",\"gall bladder\",\"brain\",\"heart\",\"lung\",\"ovary\",\"blood platelet\",\"frontal lobe\",\"B-lymphocyte\",\"colon\",\"placenta\",\"monocyte\",\"rectum\",\"esophagus\"]','[\"hair shaft\"]','[\"blood plasma\",\"blood\"]','[\"stomach\"]','[\"uterine endometrium\"]','[\"ascites\",\"macrophage\"]','[\"skin fibroblast\"]','[\"primary cell\",\"blood\"]','[\"oocyte\",\"colon\",\"brain\"]','[\"retinal microvascular endothelial cell\"]','[\"kidney\",\"blood plasma\",\"blood\"]','[\"dendritic cell\"]','[\"liver\",\"spleen\",\"skin\",\"thyroid gland\",\"stomach\",\"artery\",\"testis\",\"brain\",\"heart\",\"saliva\",\"lung\",\"ovary\",\"colon\",\"small intestine\",\"skeletal muscle\",\"esophagus\"]']"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "source": [
    "human_meta_filtered = human_meta_reprocessed[human_meta_reprocessed['tissues'].isin(keep_list)]\n",
    "print(human_meta_filtered.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(256, 15)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "# Which experiment types should NOT be included? Those that do some sort of purification and thus not all naturally occuring peptides are found in the simple\n",
    "\n",
    "These experiment types include everything with affinity purification"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "source": [
    "bad_types = ['[\"Affinity purification coupled with mass spectrometry proteomics\",\"Shotgun proteomics\",\"Gel-based experiment\"]',\n",
    "'[\"Affinity purification coupled with mass spectrometry proteomics\",\"Shotgun proteomics\"]',\n",
    "'[\"Affinity purification coupled with mass spectrometry proteomics\",\"Top-down proteomics\",\"Shotgun proteomics\"]',\n",
    "'[\"Affinity purification coupled with mass spectrometry proteomics\"]',\n",
    "'Unknown']\n",
    "human_meta_filtered_shotgun = human_meta_filtered[~human_meta_filtered['experimentTypes'].isin(bad_types)]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "source": [
    "print(human_meta_filtered_shotgun.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(239, 15)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "source": [
    "shotgun_pxds = human_meta_filtered_shotgun['accession'].tolist()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "source": [
    "human_meta_filtered_shotgun.to_csv('metadata_human_shotgun.csv', index=False, sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## This file contains the PRIDE metadata from the selected projects, these projects need to be annotated at the raw file level"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "source": [
    "projects = human_meta_filtered_shotgun[['accession', 'tissues']]\n",
    "selected_pxds = projects['accession'].tolist()\n",
    "projects.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>tissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD003411</td>\n",
       "      <td>[\"umbilical vein\"]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PXD000072</td>\n",
       "      <td>[\"blood platelet\"]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PXD003409</td>\n",
       "      <td>[\"umbilical vein\"]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PXD003534</td>\n",
       "      <td>[\"blood\"]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PXD001674</td>\n",
       "      <td>[\"blood\"]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    accession             tissues\n",
       "0   PXD003411  [\"umbilical vein\"]\n",
       "14  PXD000072  [\"blood platelet\"]\n",
       "15  PXD003409  [\"umbilical vein\"]\n",
       "19  PXD003534           [\"blood\"]\n",
       "20  PXD001674           [\"blood\"]"
      ]
     },
     "metadata": {},
     "execution_count": 48
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "source": [
    "project_tup = [tuple(x) for x in projects.values]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "pxd_assay = pd.DataFrame(columns=['accession', 'tissues', 'assay_name'])\n",
    "for i in project_tup:\n",
    "    p = i[0]\n",
    "    t = i[1]\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/*.mgf.ionbot.csv'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/*.mgf.ionbot.csv'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.3/*.mgf.ionbot.csv'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0] # split so you just get filename and not the path\n",
    "            insert_row(pxd_assay, [p,t,assay_name])\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0]\n",
    "            insert_row(pxd_assay, [p,t,assay_name])\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0]\n",
    "            insert_row(pxd_assay, [p,t,assay_name])"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "pxd_assay.shape"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(13516, 3)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "pxd_assay.to_csv('PXD_tissue_assay_mapping.csv', sep='\\t', index=False)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Previously, manual annotation was performed. This annotation will be added"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "source": [
    "manual = pd.read_csv('manual_tissue_annotation_shotgun_22092021.csv', sep=';')\n",
    "manual.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>assay_name</th>\n",
       "      <th>pride_tissue</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>tissue</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>HSE3</td>\n",
       "      <td>['brain']</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>HSA</td>\n",
       "      <td>['brain']</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>HSA3</td>\n",
       "      <td>['brain']</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>HSE</td>\n",
       "      <td>['brain']</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PXD000004</td>\n",
       "      <td>HSB3</td>\n",
       "      <td>['brain']</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Brain</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   accession assay_name pride_tissue cell_type tissue   status\n",
       "0  PXD000004       HSE3    ['brain']     Brain  Brain  Healthy\n",
       "1  PXD000004        HSA    ['brain']     Brain  Brain  Healthy\n",
       "2  PXD000004       HSA3    ['brain']     Brain  Brain  Healthy\n",
       "3  PXD000004        HSE    ['brain']     Brain  Brain  Healthy\n",
       "4  PXD000004       HSB3    ['brain']     Brain  Brain  Healthy"
      ]
     },
     "metadata": {},
     "execution_count": 53
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "source": [
    "y = manual[manual['accession'].isin(selected_pxds)]\n",
    "print(len(np.unique(y['accession'])))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "217\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "There are projectst in selected pxds that are not in the manual annotation file!"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "source": [
    "y = set(y['accession'].tolist())"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "source": [
    "set(selected_pxds) - set(y)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "{'PXD005978',\n",
       " 'PXD008870',\n",
       " 'PXD010705',\n",
       " 'PXD011116',\n",
       " 'PXD011129',\n",
       " 'PXD012029',\n",
       " 'PXD012615',\n",
       " 'PXD013038',\n",
       " 'PXD014524',\n",
       " 'PXD014894',\n",
       " 'PXD015163',\n",
       " 'PXD015518',\n",
       " 'PXD015780',\n",
       " 'PXD015817',\n",
       " 'PXD016126',\n",
       " 'PXD016999',\n",
       " 'PXD017011',\n",
       " 'PXD017056',\n",
       " 'PXD017656',\n",
       " 'PXD018208',\n",
       " 'PXD018226',\n",
       " 'PXD018502'}"
      ]
     },
     "metadata": {},
     "execution_count": 69
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "source": [
    "len(set(selected_pxds) - set(y))"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "22"
      ]
     },
     "metadata": {},
     "execution_count": 70
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "source": [
    "missing_projects = set(selected_pxds) - set(y) #number of projects that still need manual annotation\n",
    "missing_project_annotation = human_meta_filtered_shotgun[human_meta_filtered_shotgun['accession'].isin(missing_projects)]\n",
    "missing_project_annotation = missing_project_annotation[['accession','tissues' ]]\n",
    "missing_project_annotation.shape"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "(22, 2)"
      ]
     },
     "metadata": {},
     "execution_count": 71
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "source": [
    "missing_project_annotation_tup = [tuple(x) for x in missing_project_annotation.values]\n",
    "pxd_assay = pd.DataFrame(columns=['accession', 'tissues', 'assay_name'])\n",
    "for i in missing_project_annotation_tup:\n",
    "    p = i[0]\n",
    "    t = i[1]\n",
    "    print(p, t)\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/*.mgf.ionbot.csv'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/*.mgf.ionbot.csv'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.3/*.mgf.ionbot.csv'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0] # split so you just get filename and not the path\n",
    "            insert_row(pxd_assay, [p,t,assay_name])\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0]\n",
    "            insert_row(pxd_assay, [p,t,assay_name])\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.getsize(file) != 0: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            assay_name = (str(file).split('/')[-1]).split('.')[0]\n",
    "            insert_row(pxd_assay, [p,t,assay_name])"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "PXD008870 [\"feces\"]\n",
      "PXD010705 [\"cerebrospinal fluid\"]\n",
      "PXD018226 [\"cerebrospinal fluid\"]\n",
      "PXD018502 [\"blood serum\"]\n",
      "PXD016126 [\"saliva\"]\n",
      "PXD015817 [\"dental plaque\"]\n",
      "PXD017011 [\"kidney\"]\n",
      "PXD015780 [\"blood\"]\n",
      "PXD017656 [\"brain\"]\n",
      "PXD005978 [\"blood\"]\n",
      "PXD012615 [\"liver\"]\n",
      "PXD017056 [\"blood\"]\n",
      "PXD014894 [\"blood\"]\n",
      "PXD015163 [\"blood plasma\"]\n",
      "PXD018208 [\"blood\"]\n",
      "PXD013038 [\"kidney\"]\n",
      "PXD011116 [\"lung\"]\n",
      "PXD011129 [\"lung\"]\n",
      "PXD015518 [\"skin\"]\n",
      "PXD012029 [\"colon\"]\n",
      "PXD014524 [\"kidney\",\"blood plasma\",\"blood\"]\n",
      "PXD016999 [\"liver\",\"spleen\",\"skin\",\"thyroid gland\",\"stomach\",\"artery\",\"testis\",\"brain\",\"heart\",\"saliva\",\"lung\",\"ovary\",\"colon\",\"small intestine\",\"skeletal muscle\",\"esophagus\"]\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "source": [
    "pxd_assay.head(12)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>tissues</th>\n",
       "      <th>assay_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>NO_36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>PD_53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>PD_39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>PD_26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>NO_38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>NO_25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PXD008870</td>\n",
       "      <td>[\"feces\"]</td>\n",
       "      <td>CO_10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    accession    tissues assay_name\n",
       "0   PXD008870  [\"feces\"]      CO_03\n",
       "1   PXD008870  [\"feces\"]      NO_36\n",
       "2   PXD008870  [\"feces\"]      PD_53\n",
       "3   PXD008870  [\"feces\"]      PD_39\n",
       "4   PXD008870  [\"feces\"]      PD_26\n",
       "5   PXD008870  [\"feces\"]      CO_20\n",
       "6   PXD008870  [\"feces\"]      CO_13\n",
       "7   PXD008870  [\"feces\"]      CO_01\n",
       "8   PXD008870  [\"feces\"]      NO_38\n",
       "9   PXD008870  [\"feces\"]      CO_19\n",
       "10  PXD008870  [\"feces\"]      NO_25\n",
       "11  PXD008870  [\"feces\"]      CO_10"
      ]
     },
     "metadata": {},
     "execution_count": 74
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "source": [
    "pxd_assay.to_csv('PXD_tissue_assay_mapping_shotgun.csv', sep='\\t', index=False)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "After manual annotation, the newly annotated pxds are added to the original manual annotation and the new file is called **manual_tissue_annotation_shotgun_22092021.csv**\n",
    "\n",
    "However, the original file contained non-shotgun experiments so these need to be removed!"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "source": [
    "print(len(selected_pxds))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "239\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "source": [
    "annotated_file = pd.read_csv('manual_tissue_annotation_shotgun_22092021.csv', sep=';')\n",
    "print(annotated_file.shape)\n",
    "annotated_file.tail()"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(14244, 6)\n"
     ]
    },
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>assay_name</th>\n",
       "      <th>pride_tissue</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>tissue</th>\n",
       "      <th>status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14239</th>\n",
       "      <td>PXD009261</td>\n",
       "      <td>Cornea_IG_6</td>\n",
       "      <td>[\"cornea\"]</td>\n",
       "      <td>Eye</td>\n",
       "      <td>Sclera</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14240</th>\n",
       "      <td>PXD009261</td>\n",
       "      <td>Cornea_bRPLC_22</td>\n",
       "      <td>[\"cornea\"]</td>\n",
       "      <td>Eye</td>\n",
       "      <td>Sclera</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14241</th>\n",
       "      <td>PXD009261</td>\n",
       "      <td>Cornea_IG_17</td>\n",
       "      <td>[\"cornea\"]</td>\n",
       "      <td>Eye</td>\n",
       "      <td>Sclera</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14242</th>\n",
       "      <td>PXD009261</td>\n",
       "      <td>Cornea_IG_16</td>\n",
       "      <td>[\"cornea\"]</td>\n",
       "      <td>Eye</td>\n",
       "      <td>Sclera</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14243</th>\n",
       "      <td>PXD009261</td>\n",
       "      <td>Cornea_IG_18</td>\n",
       "      <td>[\"cornea\"]</td>\n",
       "      <td>Eye</td>\n",
       "      <td>Sclera</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       accession       assay_name pride_tissue cell_type  tissue   status\n",
       "14239  PXD009261      Cornea_IG_6   [\"cornea\"]       Eye  Sclera  Healthy\n",
       "14240  PXD009261  Cornea_bRPLC_22   [\"cornea\"]       Eye  Sclera  Healthy\n",
       "14241  PXD009261     Cornea_IG_17   [\"cornea\"]       Eye  Sclera  Healthy\n",
       "14242  PXD009261     Cornea_IG_16   [\"cornea\"]       Eye  Sclera  Healthy\n",
       "14243  PXD009261     Cornea_IG_18   [\"cornea\"]       Eye  Sclera  Healthy"
      ]
     },
     "metadata": {},
     "execution_count": 77
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "source": [
    "annotated_shotgun_file = annotated_file[annotated_file['accession'].isin(selected_pxds)]\n",
    "print(annotated_shotgun_file.shape)\n",
    "annotated_shotgun_file.to_csv('manual_tissue_annotation_shotgun_checked_22092021.csv', index=False, sep=';')"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(11860, 6)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [],
   "outputs": [],
   "metadata": {}
  }
 ]
}