{
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  },
  "orig_nbformat": 2,
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.6.13 64-bit ('tissuespecific': conda)"
  },
  "interpreter": {
   "hash": "26aff48bbd2e903738c4a95289dc9f6861d7c456d77b24504d90c24e131deb55"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2,
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob\n",
    "import os\n",
    "from collections import defaultdict\n",
    "def insert_row(df, my_row):\n",
    "    df.loc[len(df)] = my_row"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "meta = pd.read_csv('metadata_ionbot_update_0401.csv')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "human_meta = meta[meta['species'] == '[\"Homo sapiens (Human)\"]']\n",
    "print(human_meta.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(1347, 15)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "source": [
    "human_meta.drop(human_meta.loc[human_meta['tissues']== '[\"cell culture\"]'].index, inplace=True)\n",
    "print(human_meta.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(946, 15)\n",
      "/home/compomics/git/tissue-prediction/venv/lib/python3.6/site-packages/pandas/core/frame.py:4174: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "hpa_tissues = ['adipose tissue','adrenal gland','amygdala','appendix','b-cells','basal ganglia','bone marrow','breast','cerebellum','cerebral cortex','uterine cervix','colon','corpus callosum','dendritic cells','ductus deferens','duodenum','endometrium','epididymis','esophagus','fallopian tube','gall bladder','granulocytes','heart','hippocampal formation','hypothalamus','kidney','liver','lung','lymph node','midbrain','monocytes','nk-cells','olfactory region','ovary','pancreas','parathyroid gland','pituitary gland','placenta','pons and medulla','prostate','rectum','retina','salivary gland','seminal vesicle','skeletal muscle','skin','small intestine','smooth muscle','spinal cord','spleen','stomach','t-cells','testis','thalamus','thymus','thyroid gland','tongue','tonsil','peripheral blood mononuclear cell','urinary bladder','vagina']"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "source": [
    "pride_tissues = list(human_meta['tissues'].unique())\n",
    "pride_tissues = [x for x in pride_tissues if str(x) != 'nan']"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    "tissues_dict = defaultdict(list)\n",
    "for i in hpa_tissues:\n",
    "    x = []\n",
    "    for s in pride_tissues:\n",
    "        if i.lower() in s.lower():\n",
    "            x.append(s)\n",
    "    tissues_dict[i] = x"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "source": [
    "not_found = []\n",
    "found = []\n",
    "for key, value in tissues_dict.items():\n",
    "    if len(value) == 0:\n",
    "        not_found.append(key)\n",
    "    else:\n",
    "        found.append(key)\n",
    "    "
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "source": [
    "print(not_found)\n",
    "print(found)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "['amygdala', 'b-cells', 'basal ganglia', 'cerebellum', 'corpus callosum', 'dendritic cells', 'ductus deferens', 'epididymis', 'fallopian tube', 'granulocytes', 'hippocampal formation', 'midbrain', 'monocytes', 'nk-cells', 'olfactory region', 'parathyroid gland', 'pituitary gland', 'pons and medulla', 'salivary gland', 'seminal vesicle', 'smooth muscle', 't-cells', 'thymus']\n",
      "['adipose tissue', 'adrenal gland', 'appendix', 'bone marrow', 'breast', 'cerebral cortex', 'uterine cervix', 'colon', 'duodenum', 'endometrium', 'esophagus', 'gall bladder', 'heart', 'hypothalamus', 'kidney', 'liver', 'lung', 'lymph node', 'ovary', 'pancreas', 'placenta', 'prostate', 'rectum', 'retina', 'skeletal muscle', 'skin', 'small intestine', 'spinal cord', 'spleen', 'stomach', 'testis', 'thalamus', 'thyroid gland', 'tongue', 'tonsil', 'peripheral blood mononuclear cell', 'urinary bladder', 'vagina']\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "T cells, monocytes, b cells, nk cells are found in the Pandey dataset with accession nummer PXD000561\n"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "source": [
    "human_meta.to_csv('metadata_pride_human_no_cellculture.csv', sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "human_meta= pd.read_csv('metadata_pride_human_no_cellculture.csv', sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "source": [
    "human_meta.columns"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'accession', 'url', 'title', 'projectDescription',\n",
       "       'species', 'tissues', 'ptms', 'instrumentNames', 'projectTags',\n",
       "       'experimentTypes', 'quantificationMethods', 'keywords',\n",
       "       'dataProcessingProtocol', 'submissionDate', 'references'],\n",
       "      dtype='object')"
      ]
     },
     "metadata": {},
     "execution_count": 8
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "source": [
    "human_meta.shape"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "(946, 16)"
      ]
     },
     "metadata": {},
     "execution_count": 18
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "source": [
    "selected_datasets = pd.DataFrame(columns=['accession', 'url', 'title', 'projectDescription', 'species', 'tissues',\n",
    "       'ptms', 'instrumentNames', 'projectTags', 'experimentTypes',\n",
    "       'quantificationMethods', 'keywords', 'dataProcessingProtocol',\n",
    "       'submissionDate', 'references', 'hpa_tissue'])\n",
    "for key, values in tissues_dict.items():\n",
    "    for v in values:\n",
    "        row = pd.DataFrame(human_meta[human_meta['tissues'] == v])\n",
    "        row['hpa_tissue'] = key\n",
    "        selected_datasets = pd.concat([selected_datasets, row],axis=0)\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "source": [
    "selected_datasets"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>url</th>\n",
       "      <th>title</th>\n",
       "      <th>projectDescription</th>\n",
       "      <th>species</th>\n",
       "      <th>tissues</th>\n",
       "      <th>ptms</th>\n",
       "      <th>instrumentNames</th>\n",
       "      <th>projectTags</th>\n",
       "      <th>experimentTypes</th>\n",
       "      <th>quantificationMethods</th>\n",
       "      <th>keywords</th>\n",
       "      <th>dataProcessingProtocol</th>\n",
       "      <th>submissionDate</th>\n",
       "      <th>references</th>\n",
       "      <th>hpa_tissue</th>\n",
       "      <th>Unnamed: 0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>638</th>\n",
       "      <td>PXD009893</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Proteomic profiling of human mesenchymal stem ...</td>\n",
       "      <td>We performed a proteomic profiling of human me...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"adipose tissue\",\"umbilical cord\"]</td>\n",
       "      <td>[{\"ptm\":\"No PTMs are included in the dataset\"}]</td>\n",
       "      <td>[\"TripleTOF 5600\",\"Q Exactive HF\"]</td>\n",
       "      <td>[\"Human Proteome Project\",\"Biological\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Mesenchymal Stem Cells, hMSCs, Adipose T...</td>\n",
       "      <td>We searched the acquired MS/MS spectra against...</td>\n",
       "      <td>2018-05-25</td>\n",
       "      <td>[[\"DOI:10.1021/acs.jproteome.8b00384\",\"PMID:29...</td>\n",
       "      <td>adipose tissue</td>\n",
       "      <td>944.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>658</th>\n",
       "      <td>PXD008146</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>secretome of human chondrocytes and adipose me...</td>\n",
       "      <td>Objective: Assuming that mesenchymal stem cell...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"articular cartilage\",\"cell culture\",\"abdomin...</td>\n",
       "      <td>[{\"ptm\":\"No PTMs are included in the dataset\"}]</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"Biological\"]</td>\n",
       "      <td>[\"Gel-based experiment\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human, chondrocyte, adipose, mesenchymal stem ...</td>\n",
       "      <td>MS/MS data were searched in a Human database f...</td>\n",
       "      <td>2017-11-07</td>\n",
       "      <td>[[\"PMID:29238343\",\"DOI:10.3389/fimmu.2017.0163...</td>\n",
       "      <td>adipose tissue</td>\n",
       "      <td>971.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779</th>\n",
       "      <td>PXD000561</td>\n",
       "      <td>https://www.ebi.ac.uk:443/pride/ws/archive/pro...</td>\n",
       "      <td>A draft map of the human proteome</td>\n",
       "      <td>The availability of human genome sequence has ...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"prostate gland\",\"gut\",\"testis\",\"helper T-lym...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\",\"LTQ Orbitrap Elite\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[\"Bottom-up proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human proteome map, LC-MS/MS, proteogenomics</td>\n",
       "      <td>Mass spectrometry data obtained from all LC-MS...</td>\n",
       "      <td>2014-02-25</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:24669763\"],[\"PMID:24870542\"...</td>\n",
       "      <td>adrenal gland</td>\n",
       "      <td>1173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>684</th>\n",
       "      <td>PXD010086</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>N-glycan profiling of Pseudomyxoma peritonei</td>\n",
       "      <td>Pseudomyxoma peritonei (PMP) is a subtype of m...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"vermiform appendix\"]</td>\n",
       "      <td>[{\"ptm\":\"monosodium salt\",\"UNIMOD\":\"30\"}]</td>\n",
       "      <td>[\"ultraflex\"]</td>\n",
       "      <td>[\"Biological\"]</td>\n",
       "      <td>[\"Top-down proteomics\"]</td>\n",
       "      <td>[\"Normalized Spectral Abundance Factor - NSAF\"]</td>\n",
       "      <td>human appendix cancer N-glycan fucosylation</td>\n",
       "      <td>Relative molar abundances of neutral and acidi...</td>\n",
       "      <td>2018-06-11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>appendix</td>\n",
       "      <td>1011.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>PXD001449</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Validation of alternatively spliced protein is...</td>\n",
       "      <td>Alternative splicing of mRNA diversifies the f...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"bone marrow\"]</td>\n",
       "      <td>[{\"ptm\":\"iodoacetamide derivatized residue\",\"U...</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human, mesenchymal stem cell, LC-MS/MS</td>\n",
       "      <td>Proteolytic peptide samples were separated by ...</td>\n",
       "      <td>2014-10-31</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:25961807\"]]</td>\n",
       "      <td>bone marrow</td>\n",
       "      <td>292.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>PXD004249</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Human bladder cancer and colon cancer quatitat...</td>\n",
       "      <td>We analyzed a set of 40 independently collecte...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"urinary bladder\",\"colon\"]</td>\n",
       "      <td>[{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human, bladder cancer, colon cancer, adenocarc...</td>\n",
       "      <td>MS/MS spectra were extracted from raw files us...</td>\n",
       "      <td>2016-05-31</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:27927741\"]]</td>\n",
       "      <td>urinary bladder</td>\n",
       "      <td>242.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>779</th>\n",
       "      <td>PXD000561</td>\n",
       "      <td>https://www.ebi.ac.uk:443/pride/ws/archive/pro...</td>\n",
       "      <td>A draft map of the human proteome</td>\n",
       "      <td>The availability of human genome sequence has ...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"prostate gland\",\"gut\",\"testis\",\"helper T-lym...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\",\"LTQ Orbitrap Elite\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[\"Bottom-up proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>human proteome map, LC-MS/MS, proteogenomics</td>\n",
       "      <td>Mass spectrometry data obtained from all LC-MS...</td>\n",
       "      <td>2014-02-25</td>\n",
       "      <td>[[\"DOI:null\",\"PMID:24669763\"],[\"PMID:24870542\"...</td>\n",
       "      <td>urinary bladder</td>\n",
       "      <td>1173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>PXD009723</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Genital proteins in women at risk of HIV infec...</td>\n",
       "      <td>In this study, we used a bead-based affinity p...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cervicovaginal fluid\"]</td>\n",
       "      <td>[{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cervicovaginal secretions</td>\n",
       "      <td>Equal amounts of sample peptides were analyzed...</td>\n",
       "      <td>2018-05-09</td>\n",
       "      <td>[[\"PMID:30504243\",\"DOI:10.1074/mcp.ra118.00075...</td>\n",
       "      <td>vagina</td>\n",
       "      <td>185.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>PXD001435</td>\n",
       "      <td>https://www.ebi.ac.uk/pride/ws/archive/project...</td>\n",
       "      <td>Cervicovaginal dysbiosis is associated with si...</td>\n",
       "      <td>Cervicovaginal microbiome dysbiosis is associa...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cervicovaginal fluid\"]</td>\n",
       "      <td>[{\"ptm\":\"iodoacetamide derivatized residue\",\"U...</td>\n",
       "      <td>[\"LTQ Orbitrap Velos\"]</td>\n",
       "      <td>[\"Biological\",\"Biomedical\"]</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>[\"MS1 intensity based label-free quantificatio...</td>\n",
       "      <td>bacterial vaginosis/ cervicovaginal microbiome...</td>\n",
       "      <td>Thermo RAW files were imported into Progenesis...</td>\n",
       "      <td>2014-10-27</td>\n",
       "      <td>[[\"DOI:10.1038/mi.2015.86\",\"PMID:26349657\"]]</td>\n",
       "      <td>vagina</td>\n",
       "      <td>313.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>780</th>\n",
       "      <td>PXD014300</td>\n",
       "      <td>https://www.ebi.ac.uk:443/pride/ws/archive/pro...</td>\n",
       "      <td>Chr18 Consortium MS-data for missing proteins ...</td>\n",
       "      <td>Despite direct or indirect efforts of proteomi...</td>\n",
       "      <td>[\"Homo sapiens (Human)\"]</td>\n",
       "      <td>[\"cervicovaginal fluid\",\"CACO-2 cell\",\"cardiom...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[\"LTQ Orbitrap Elite\",\"Q Exactive\"]</td>\n",
       "      <td>[\"Chromosome-centric Human Proteome Project (C...</td>\n",
       "      <td>[\"Shotgun proteomics\"]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Human Proteome, Missing proteins, Uncertain pr...</td>\n",
       "      <td>Data files obtained after data-dependent shotg...</td>\n",
       "      <td>2019-06-19</td>\n",
       "      <td>[[\"PMID:32456206\",\"DOI:10.3390/proteomes802001...</td>\n",
       "      <td>vagina</td>\n",
       "      <td>1952.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>322 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     accession                                                url  \\\n",
       "638  PXD009893  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "658  PXD008146  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "779  PXD000561  https://www.ebi.ac.uk:443/pride/ws/archive/pro...   \n",
       "684  PXD010086  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "212  PXD001449  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "..         ...                                                ...   \n",
       "172  PXD004249  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "779  PXD000561  https://www.ebi.ac.uk:443/pride/ws/archive/pro...   \n",
       "133  PXD009723  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "227  PXD001435  https://www.ebi.ac.uk/pride/ws/archive/project...   \n",
       "780  PXD014300  https://www.ebi.ac.uk:443/pride/ws/archive/pro...   \n",
       "\n",
       "                                                 title  \\\n",
       "638  Proteomic profiling of human mesenchymal stem ...   \n",
       "658  secretome of human chondrocytes and adipose me...   \n",
       "779                  A draft map of the human proteome   \n",
       "684       N-glycan profiling of Pseudomyxoma peritonei   \n",
       "212  Validation of alternatively spliced protein is...   \n",
       "..                                                 ...   \n",
       "172  Human bladder cancer and colon cancer quatitat...   \n",
       "779                  A draft map of the human proteome   \n",
       "133  Genital proteins in women at risk of HIV infec...   \n",
       "227  Cervicovaginal dysbiosis is associated with si...   \n",
       "780  Chr18 Consortium MS-data for missing proteins ...   \n",
       "\n",
       "                                    projectDescription  \\\n",
       "638  We performed a proteomic profiling of human me...   \n",
       "658  Objective: Assuming that mesenchymal stem cell...   \n",
       "779  The availability of human genome sequence has ...   \n",
       "684  Pseudomyxoma peritonei (PMP) is a subtype of m...   \n",
       "212  Alternative splicing of mRNA diversifies the f...   \n",
       "..                                                 ...   \n",
       "172  We analyzed a set of 40 independently collecte...   \n",
       "779  The availability of human genome sequence has ...   \n",
       "133  In this study, we used a bead-based affinity p...   \n",
       "227  Cervicovaginal microbiome dysbiosis is associa...   \n",
       "780  Despite direct or indirect efforts of proteomi...   \n",
       "\n",
       "                      species  \\\n",
       "638  [\"Homo sapiens (Human)\"]   \n",
       "658  [\"Homo sapiens (Human)\"]   \n",
       "779  [\"Homo sapiens (Human)\"]   \n",
       "684  [\"Homo sapiens (Human)\"]   \n",
       "212  [\"Homo sapiens (Human)\"]   \n",
       "..                        ...   \n",
       "172  [\"Homo sapiens (Human)\"]   \n",
       "779  [\"Homo sapiens (Human)\"]   \n",
       "133  [\"Homo sapiens (Human)\"]   \n",
       "227  [\"Homo sapiens (Human)\"]   \n",
       "780  [\"Homo sapiens (Human)\"]   \n",
       "\n",
       "                                               tissues  \\\n",
       "638                [\"adipose tissue\",\"umbilical cord\"]   \n",
       "658  [\"articular cartilage\",\"cell culture\",\"abdomin...   \n",
       "779  [\"prostate gland\",\"gut\",\"testis\",\"helper T-lym...   \n",
       "684                             [\"vermiform appendix\"]   \n",
       "212                                    [\"bone marrow\"]   \n",
       "..                                                 ...   \n",
       "172                        [\"urinary bladder\",\"colon\"]   \n",
       "779  [\"prostate gland\",\"gut\",\"testis\",\"helper T-lym...   \n",
       "133                           [\"cervicovaginal fluid\"]   \n",
       "227                           [\"cervicovaginal fluid\"]   \n",
       "780  [\"cervicovaginal fluid\",\"CACO-2 cell\",\"cardiom...   \n",
       "\n",
       "                                                  ptms  \\\n",
       "638    [{\"ptm\":\"No PTMs are included in the dataset\"}]   \n",
       "658    [{\"ptm\":\"No PTMs are included in the dataset\"}]   \n",
       "779                                                NaN   \n",
       "684          [{\"ptm\":\"monosodium salt\",\"UNIMOD\":\"30\"}]   \n",
       "212  [{\"ptm\":\"iodoacetamide derivatized residue\",\"U...   \n",
       "..                                                 ...   \n",
       "172  [{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...   \n",
       "779                                                NaN   \n",
       "133  [{\"ptm\":\"Carbamidomethyl\",\"UNIMOD\":\"4\"},{\"ptm\"...   \n",
       "227  [{\"ptm\":\"iodoacetamide derivatized residue\",\"U...   \n",
       "780                                                NaN   \n",
       "\n",
       "                                 instrumentNames  \\\n",
       "638           [\"TripleTOF 5600\",\"Q Exactive HF\"]   \n",
       "658                       [\"LTQ Orbitrap Velos\"]   \n",
       "779  [\"LTQ Orbitrap Velos\",\"LTQ Orbitrap Elite\"]   \n",
       "684                                [\"ultraflex\"]   \n",
       "212                       [\"LTQ Orbitrap Velos\"]   \n",
       "..                                           ...   \n",
       "172                       [\"LTQ Orbitrap Velos\"]   \n",
       "779  [\"LTQ Orbitrap Velos\",\"LTQ Orbitrap Elite\"]   \n",
       "133                       [\"LTQ Orbitrap Velos\"]   \n",
       "227                       [\"LTQ Orbitrap Velos\"]   \n",
       "780          [\"LTQ Orbitrap Elite\",\"Q Exactive\"]   \n",
       "\n",
       "                                           projectTags  \\\n",
       "638            [\"Human Proteome Project\",\"Biological\"]   \n",
       "658                                     [\"Biological\"]   \n",
       "779                                                NaN   \n",
       "684                                     [\"Biological\"]   \n",
       "212                                     [\"Biomedical\"]   \n",
       "..                                                 ...   \n",
       "172                                     [\"Biomedical\"]   \n",
       "779                                                NaN   \n",
       "133                                     [\"Biomedical\"]   \n",
       "227                        [\"Biological\",\"Biomedical\"]   \n",
       "780  [\"Chromosome-centric Human Proteome Project (C...   \n",
       "\n",
       "              experimentTypes  \\\n",
       "638    [\"Shotgun proteomics\"]   \n",
       "658  [\"Gel-based experiment\"]   \n",
       "779  [\"Bottom-up proteomics\"]   \n",
       "684   [\"Top-down proteomics\"]   \n",
       "212    [\"Shotgun proteomics\"]   \n",
       "..                        ...   \n",
       "172    [\"Shotgun proteomics\"]   \n",
       "779  [\"Bottom-up proteomics\"]   \n",
       "133    [\"Shotgun proteomics\"]   \n",
       "227    [\"Shotgun proteomics\"]   \n",
       "780    [\"Shotgun proteomics\"]   \n",
       "\n",
       "                                 quantificationMethods  \\\n",
       "638                                                NaN   \n",
       "658                                                NaN   \n",
       "779                                                NaN   \n",
       "684    [\"Normalized Spectral Abundance Factor - NSAF\"]   \n",
       "212                                                NaN   \n",
       "..                                                 ...   \n",
       "172                                                NaN   \n",
       "779                                                NaN   \n",
       "133                                                NaN   \n",
       "227  [\"MS1 intensity based label-free quantificatio...   \n",
       "780                                                NaN   \n",
       "\n",
       "                                              keywords  \\\n",
       "638  Human Mesenchymal Stem Cells, hMSCs, Adipose T...   \n",
       "658  human, chondrocyte, adipose, mesenchymal stem ...   \n",
       "779       human proteome map, LC-MS/MS, proteogenomics   \n",
       "684        human appendix cancer N-glycan fucosylation   \n",
       "212             Human, mesenchymal stem cell, LC-MS/MS   \n",
       "..                                                 ...   \n",
       "172  human, bladder cancer, colon cancer, adenocarc...   \n",
       "779       human proteome map, LC-MS/MS, proteogenomics   \n",
       "133                          Cervicovaginal secretions   \n",
       "227  bacterial vaginosis/ cervicovaginal microbiome...   \n",
       "780  Human Proteome, Missing proteins, Uncertain pr...   \n",
       "\n",
       "                                dataProcessingProtocol submissionDate  \\\n",
       "638  We searched the acquired MS/MS spectra against...     2018-05-25   \n",
       "658  MS/MS data were searched in a Human database f...     2017-11-07   \n",
       "779  Mass spectrometry data obtained from all LC-MS...     2014-02-25   \n",
       "684  Relative molar abundances of neutral and acidi...     2018-06-11   \n",
       "212  Proteolytic peptide samples were separated by ...     2014-10-31   \n",
       "..                                                 ...            ...   \n",
       "172  MS/MS spectra were extracted from raw files us...     2016-05-31   \n",
       "779  Mass spectrometry data obtained from all LC-MS...     2014-02-25   \n",
       "133  Equal amounts of sample peptides were analyzed...     2018-05-09   \n",
       "227  Thermo RAW files were imported into Progenesis...     2014-10-27   \n",
       "780  Data files obtained after data-dependent shotg...     2019-06-19   \n",
       "\n",
       "                                            references       hpa_tissue  \\\n",
       "638  [[\"DOI:10.1021/acs.jproteome.8b00384\",\"PMID:29...   adipose tissue   \n",
       "658  [[\"PMID:29238343\",\"DOI:10.3389/fimmu.2017.0163...   adipose tissue   \n",
       "779  [[\"DOI:null\",\"PMID:24669763\"],[\"PMID:24870542\"...    adrenal gland   \n",
       "684                                                NaN         appendix   \n",
       "212                     [[\"DOI:null\",\"PMID:25961807\"]]      bone marrow   \n",
       "..                                                 ...              ...   \n",
       "172                     [[\"DOI:null\",\"PMID:27927741\"]]  urinary bladder   \n",
       "779  [[\"DOI:null\",\"PMID:24669763\"],[\"PMID:24870542\"...  urinary bladder   \n",
       "133  [[\"PMID:30504243\",\"DOI:10.1074/mcp.ra118.00075...           vagina   \n",
       "227       [[\"DOI:10.1038/mi.2015.86\",\"PMID:26349657\"]]           vagina   \n",
       "780  [[\"PMID:32456206\",\"DOI:10.3390/proteomes802001...           vagina   \n",
       "\n",
       "     Unnamed: 0  \n",
       "638       944.0  \n",
       "658       971.0  \n",
       "779      1173.0  \n",
       "684      1011.0  \n",
       "212       292.0  \n",
       "..          ...  \n",
       "172       242.0  \n",
       "779      1173.0  \n",
       "133       185.0  \n",
       "227       313.0  \n",
       "780      1952.0  \n",
       "\n",
       "[322 rows x 17 columns]"
      ]
     },
     "metadata": {},
     "execution_count": 17
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "source": [
    "#find if all these pxds are in the conodes\n",
    "PXDs = selected_datasets[['accession']].values"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "source": [
    "reprocessed = 0\n",
    "reprocessed_projects = []\n",
    "for i in PXDs:\n",
    "    p = ''.join(list(i))\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 53'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 54'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 55'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "\n",
    "print(reprocessed)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "PXD009893 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD006419 was found in conode 54\n",
      "PXD009738 was found in conode 53\n",
      "PXD001325 was found in conode 53\n",
      "PXD001326 was found in conode 53\n",
      "PXD001324 was found in conode 53\n",
      "PXD001327 was found in conode 53\n",
      "PXD008012 was found in conode 54\n",
      "PXD009570 was found in conode 54\n",
      "PXD008795 was found in conode 53\n",
      "PXD008723 was found in conode 53\n",
      "PXD004796 was found in conode 53\n",
      "PXD004778 was found in conode 53\n",
      "PXD004788 was found in conode 53\n",
      "PXD005389 was found in conode 53\n",
      "PXD004764 was found in conode 53\n",
      "PXD000445 was found in conode 53\n",
      "PXD008440 was found in conode 53\n",
      "PXD006776 was found in conode 53\n",
      "PXD007959 was found in conode 54\n",
      "PXD007740 was found in conode 54\n",
      "PXD005693 was found in conode 54\n",
      "PXD008147 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD007231 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD008205 was found in conode 53\n",
      "PXD008934 was found in conode 53\n",
      "PXD006911 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD008795 was found in conode 53\n",
      "PXD006522 was found in conode 53\n",
      "PXD006897 was found in conode 53\n",
      "PXD006517 was found in conode 53\n",
      "PXD007080 was found in conode 54\n",
      "PXD002133 was found in conode 54\n",
      "PXD007081 was found in conode 54\n",
      "PXD009300 was found in conode 54\n",
      "PXD008824 was found in conode 54\n",
      "PXD010142 was found in conode 53\n",
      "PXD009476 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD006833 was found in conode 53\n",
      "PXD000533 was found in conode 53\n",
      "PXD009021 was found in conode 53\n",
      "PXD007258 was found in conode 53\n",
      "PXD010294 was found in conode 54\n",
      "PXD000529 was found in conode 54\n",
      "PXD005323 was found in conode 54\n",
      "PXD005709 was found in conode 54\n",
      "PXD010271 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD010787 was found in conode 53\n",
      "PXD004683 was found in conode 53\n",
      "PXD009996 was found in conode 53\n",
      "PXD009752 was found in conode 53\n",
      "PXD004682 was found in conode 53\n",
      "PXD005733 was found in conode 53\n",
      "PXD010786 was found in conode 53\n",
      "PXD007137 was found in conode 53\n",
      "PXD006095 was found in conode 53\n",
      "PXD006265 was found in conode 53\n",
      "PXD006096 was found in conode 53\n",
      "PXD009754 was found in conode 53\n",
      "PXD009755 was found in conode 53\n",
      "PXD008682 was found in conode 54\n",
      "PXD007123 was found in conode 54\n",
      "PXD010284 was found in conode 54\n",
      "PXD007122 was found in conode 54\n",
      "PXD011241 was found in conode 54\n",
      "PXD007705 was found in conode 54\n",
      "PXD010286 was found in conode 54\n",
      "PXD009916 was found in conode 54\n",
      "PXD009917 was found in conode 54\n",
      "PXD003743 was found in conode 54\n",
      "PXD009444 was found in conode 54\n",
      "PXD006419 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD002990 was found in conode 53\n",
      "PXD011152 was found in conode 54\n",
      "PXD011175 was found in conode 54\n",
      "PXD010271 was found in conode 53\n",
      "PXD008832 was found in conode 53\n",
      "PXD005055 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010150 was found in conode 53\n",
      "PXD009254 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD010630 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD009868 was found in conode 54\n",
      "PXD010527 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD010821 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD009196 was found in conode 54\n",
      "PXD007592 was found in conode 53\n",
      "PXD003413 was found in conode 53\n",
      "PXD006268 was found in conode 53\n",
      "PXD003971 was found in conode 53\n",
      "PXD003963 was found in conode 53\n",
      "PXD003966 was found in conode 53\n",
      "PXD003415 was found in conode 53\n",
      "PXD003412 was found in conode 53\n",
      "PXD003965 was found in conode 53\n",
      "PXD003967 was found in conode 53\n",
      "PXD008996 was found in conode 53\n",
      "PXD003414 was found in conode 53\n",
      "PXD003417 was found in conode 53\n",
      "PXD003416 was found in conode 53\n",
      "PXD011492 was found in conode 54\n",
      "PXD009880 was found in conode 54\n",
      "PXD011000 was found in conode 54\n",
      "PXD006743 was found in conode 54\n",
      "PXD011609 was found in conode 54\n",
      "PXD009317 was found in conode 54\n",
      "PXD008992 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD007935 was found in conode 53\n",
      "PXD010093 was found in conode 53\n",
      "PXD001524 was found in conode 53\n",
      "PXD006465 was found in conode 53\n",
      "PXD009737 was found in conode 53\n",
      "PXD007515 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "PXD008795 was found in conode 53\n",
      "PXD008493 was found in conode 53\n",
      "PXD006805 was found in conode 53\n",
      "PXD010260 was found in conode 54\n",
      "PXD000561 was found in conode 53\n",
      "134\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "source": [
    "x = set(PXDs.flatten())\n",
    "y = list(x)\n",
    "not_repr = len(list(set(y) - set(reprocessed_projects)))\n",
    "total = len(y)\n",
    "print('{} projects from the total of {} were not found in the reprocessed data'.format(not_repr, total))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "172 projects from the total of 285 were not found in the reprocessed data\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "mapping_xl = ['PXD000004','PXD000072','PXD000117','PXD000190','PXD000226','PXD000228','PXD000263','PXD000302','PXD000307','PXD000314','PXD000383','PXD000395','PXD000437','PXD000440','PXD000440','PXD000445','PXD000457','PXD000529','PXD000533','PXD000534','PXD000534','PXD000535','PXD000536','PXD000551','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000561','PXD000603','PXD000603','PXD000605','PXD000651','PXD000651','PXD000652','PXD000653','PXD000653','PXD000654','PXD000654','PXD000655','PXD000656','PXD000656','PXD000657','PXD000701','PXD000742','PXD000743','PXD000744','PXD000746','PXD000749','PXD000755','PXD000757','PXD000760','PXD000765','PXD000768','PXD000769','PXD000807','PXD000809','PXD000921','PXD000930','PXD000963','PXD001034','PXD001052','PXD001099','PXD001124','PXD001183','PXD001183','PXD001271','PXD001282','PXD001311','PXD001323','PXD001324','PXD001325','PXD001326','PXD001328','PXD001353','PXD001402','PXD001415','PXD001416','PXD001417','PXD001418','PXD001419','PXD001420','PXD001421','PXD001422','PXD001423','PXD001425','PXD001435','PXD001449','PXD001464','PXD001468','PXD001468','PXD001474','PXD001499','PXD001506','PXD001524','PXD001524','PXD001536','PXD001539','PXD001630','PXD001674','PXD001694','PXD001694','PXD001985','PXD001997','PXD002050','PXD002132','PXD002133','PXD002163','PXD002163','PXD002171','PXD002211','PXD002212','PXD002236','PXD002255','PXD002312','PXD002339','PXD002346','PXD002367','PXD002372','PXD002421','PXD002424','PXD002437','PXD002523','PXD002533','PXD002614','PXD002620','PXD002629','PXD002633','PXD002672','PXD002673','PXD002781','PXD002781','PXD002798','PXD002883','PXD002884','PXD002911','PXD002990','PXD003014','PXD003029','PXD003045','PXD003075','PXD003121','PXD003218','PXD003224','PXD003269','PXD003271','PXD003340','PXD003406','PXD003407','PXD003408','PXD003408','PXD003409','PXD003410','PXD003411','PXD003412','PXD003413','PXD003414','PXD003414','PXD003415','PXD003416','PXD003417','PXD003419','PXD003519','PXD003533','PXD003534','PXD003552','PXD003587','PXD003613','PXD003651','PXD003666','PXD003666','PXD003709','PXD003743','PXD003785','PXD003924','PXD003963','PXD003965','PXD003966','PXD003967','PXD003971','PXD004092','PXD004205','PXD004219','PXD004219','PXD004249','PXD004249','PXD004312','PXD004312','PXD004355','PXD004369','PXD004397','PXD004443','PXD004465','PXD004540','PXD004624','PXD004625','PXD004626','PXD004682','PXD004683','PXD004684','PXD004785','PXD004815','PXD004818','PXD004873','PXD004935','PXD004987','PXD004998','PXD004998','PXD005013','PXD005055','PXD005073','PXD005086','PXD005150','PXD005267','PXD005274','PXD005323','PXD005426','PXD005477','PXD005520','PXD005613','PXD005613','PXD005650','PXD005692','PXD005693','PXD005698','PXD005709','PXD005733','PXD005755','PXD005777','PXD005847','PXD005949','PXD005970','PXD005973','PXD005978','PXD005992','PXD006083','PXD006095','PXD006096','PXD006176','PXD006201','PXD006265','PXD006268','PXD006286','PXD006333','PXD006353','PXD006367','PXD006419','PXD006419','PXD006465','PXD006471','PXD006471','PXD006482','PXD006517','PXD006522','PXD006552','PXD006570','PXD006571','PXD006572','PXD006576','PXD006578','PXD006581','PXD006589','PXD006607','PXD006687','PXD006743','PXD006757','PXD006776','PXD006805','PXD006833','PXD006847','PXD006897','PXD006911','PXD007048','PXD007080','PXD007081','PXD007083','PXD007122','PXD007123','PXD007134','PXD007134','PXD007137','PXD007223','PXD007231','PXD007232','PXD007234','PXD007258','PXD007515','PXD007592','PXD007705','PXD007740','PXD007744','PXD007745','PXD007839','PXD007846','PXD007846','PXD007873','PXD007894','PXD007935','PXD007959','PXD007975','PXD007982','PXD008022','PXD008023','PXD008029','PXD008029','PXD008036','PXD008076','PXD008127','PXD008130','PXD008141','PXD008141','PXD008146','PXD008205','PXD008252','PXD008252','PXD008278','PXD008293','PXD008310','PXD008311','PXD008383','PXD008383','PXD008407','PXD008440','PXD008443','PXD008445','PXD008450','PXD008465','PXD008467','PXD008468','PXD008485','PXD008493','PXD008512','PXD008513','PXD008553','PXD008568','PXD008570','PXD008581','PXD008625','PXD008625','PXD008682','PXD008700','PXD008720','PXD008720','PXD008723','PXD008773','PXD008788','PXD008788','PXD008795','PXD008820','PXD008820','PXD008821','PXD008822','PXD008823','PXD008824','PXD008832','PXD008891','PXD008934','PXD008980','PXD008992','PXD008996','PXD009021','PXD009029','PXD009051','PXD009056','PXD009061','PXD009074','PXD009112','PXD009196','PXD009196','PXD009203','PXD009227','PXD009254','PXD009258','PXD009261','PXD009300','PXD009303','PXD009310','PXD009317','PXD009343','PXD009345','PXD009348','PXD009350','PXD009378','PXD009382','PXD009396','PXD009438','PXD009440','PXD009444','PXD009444','PXD009450','PXD009476','PXD009476','PXD009598','PXD009602','PXD009606','PXD009624','PXD009624','PXD009645','PXD009645','PXD009646','PXD009681','PXD009707','PXD009709','PXD009716','PXD009723','PXD009737','PXD009738','PXD009738','PXD009752','PXD009754','PXD009755','PXD009784','PXD009813','PXD009819','PXD009870','PXD009880','PXD009880','PXD009893','PXD009896','PXD009916','PXD009917','PXD009933','PXD009969','PXD009971','PXD009996','PXD010086','PXD010093','PXD010142','PXD010142','PXD010150','PXD010164','PXD010230','PXD010252','PXD010260','PXD010260','PXD010271','PXD010271','PXD010271','PXD010271','PXD010271','PXD010271','PXD010271','PXD010284','PXD010286','PXD010294','PXD010417','PXD010437','PXD010450','PXD010469','PXD010480','PXD010626','PXD010630','PXD010690','PXD010698','PXD010705','PXD010708','PXD010737','PXD010786','PXD010787','PXD010821','PXD010965','PXD010965','PXD011000','PXD011116','PXD011126','PXD011126','PXD011129','PXD011152','PXD011175','PXD011178','PXD011183','PXD011187','PXD011187','PXD011204','PXD011214','PXD011228','PXD011241','PXD011309','PXD011492','PXD011609','PXD011749','PXD011749','PXD011749','PXD011755','PXD012029','PXD012461','PXD012574','PXD012615','PXD012960','PXD012970','PXD012971','PXD013038','PXD013649','PXD013840','PXD014300','PXD014300','PXD014300','PXD014300','PXD014524','PXD014524','PXD014894','PXD015024','PXD015049','PXD015163','PXD015213','PXD015239','PXD015289','PXD015518','PXD015523','PXD015578','PXD015653','PXD015780','PXD015836','PXD015880','PXD016126','PXD016158','PXD016217','PXD016229','PXD016261','PXD016278','PXD016314','PXD016406','PXD016435','PXD016437','PXD016697','PXD016698','PXD016731','PXD016859','PXD016864','PXD016865','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD016999','PXD017011','PXD017023','PXD017030','PXD017051','PXD017056','PXD017149','PXD017195','PXD017201','PXD017213','PXD017217','PXD017268','PXD017269','PXD017269','PXD017269','PXD017656','PXD017710','PXD017799','PXD017804','PXD017805','PXD017968','PXD017976','PXD018021','PXD018208','PXD018226','PXD018357','PXD018369','PXD018445','PXD018502','PXD018588','PXD018678','PXD018682']"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "reprocessed = 0\n",
    "reprocessed_projects = []\n",
    "for i in mapping_xl:\n",
    "    p = ''.join(list(i))\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 53'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 54'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            print('{} was found in conode 55'.format(p))\n",
    "            reprocessed_projects.append(p)\n",
    "            reprocessed += 1\n",
    "\n",
    "print(reprocessed)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "PXD000004 was found in conode 53\n",
      "PXD000072 was found in conode 53\n",
      "PXD000228 was found in conode 53\n",
      "PXD000440 was found in conode 54\n",
      "PXD000440 was found in conode 54\n",
      "PXD000445 was found in conode 53\n",
      "PXD000529 was found in conode 54\n",
      "PXD000533 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000561 was found in conode 53\n",
      "PXD000605 was found in conode 53\n",
      "PXD001324 was found in conode 53\n",
      "PXD001325 was found in conode 53\n",
      "PXD001326 was found in conode 53\n",
      "PXD001415 was found in conode 53\n",
      "PXD001416 was found in conode 53\n",
      "PXD001417 was found in conode 53\n",
      "PXD001418 was found in conode 53\n",
      "PXD001419 was found in conode 53\n",
      "PXD001420 was found in conode 53\n",
      "PXD001421 was found in conode 53\n",
      "PXD001422 was found in conode 53\n",
      "PXD001423 was found in conode 53\n",
      "PXD001425 was found in conode 54\n",
      "PXD001524 was found in conode 53\n",
      "PXD001524 was found in conode 53\n",
      "PXD001674 was found in conode 53\n",
      "PXD001985 was found in conode 53\n",
      "PXD002132 was found in conode 54\n",
      "PXD002133 was found in conode 54\n",
      "PXD002211 was found in conode 53\n",
      "PXD002212 was found in conode 53\n",
      "PXD002437 was found in conode 54\n",
      "PXD002620 was found in conode 53\n",
      "PXD002672 was found in conode 54\n",
      "PXD002883 was found in conode 53\n",
      "PXD002990 was found in conode 53\n",
      "PXD003045 was found in conode 53\n",
      "PXD003075 was found in conode 53\n",
      "PXD003406 was found in conode 53\n",
      "PXD003407 was found in conode 53\n",
      "PXD003408 was found in conode 53\n",
      "PXD003408 was found in conode 53\n",
      "PXD003409 was found in conode 53\n",
      "PXD003410 was found in conode 53\n",
      "PXD003411 was found in conode 53\n",
      "PXD003412 was found in conode 53\n",
      "PXD003413 was found in conode 53\n",
      "PXD003414 was found in conode 53\n",
      "PXD003414 was found in conode 53\n",
      "PXD003415 was found in conode 53\n",
      "PXD003416 was found in conode 53\n",
      "PXD003417 was found in conode 53\n",
      "PXD003533 was found in conode 54\n",
      "PXD003534 was found in conode 53\n",
      "PXD003587 was found in conode 54\n",
      "PXD003666 was found in conode 53\n",
      "PXD003666 was found in conode 53\n",
      "PXD003709 was found in conode 53\n",
      "PXD003743 was found in conode 54\n",
      "PXD003963 was found in conode 53\n",
      "PXD003965 was found in conode 53\n",
      "PXD003966 was found in conode 53\n",
      "PXD003967 was found in conode 53\n",
      "PXD003971 was found in conode 53\n",
      "PXD004355 was found in conode 53\n",
      "PXD004443 was found in conode 54\n",
      "PXD004540 was found in conode 53\n",
      "PXD004624 was found in conode 53\n",
      "PXD004625 was found in conode 53\n",
      "PXD004626 was found in conode 53\n",
      "PXD004682 was found in conode 53\n",
      "PXD004683 was found in conode 53\n",
      "PXD005055 was found in conode 53\n",
      "PXD005323 was found in conode 54\n",
      "PXD005477 was found in conode 53\n",
      "PXD005650 was found in conode 54\n",
      "PXD005693 was found in conode 54\n",
      "PXD005709 was found in conode 54\n",
      "PXD005733 was found in conode 53\n",
      "PXD005755 was found in conode 53\n",
      "PXD006095 was found in conode 53\n",
      "PXD006096 was found in conode 53\n",
      "PXD006176 was found in conode 53\n",
      "PXD006201 was found in conode 53\n",
      "PXD006265 was found in conode 53\n",
      "PXD006268 was found in conode 53\n",
      "PXD006419 was found in conode 54\n",
      "PXD006419 was found in conode 54\n",
      "PXD006465 was found in conode 53\n",
      "PXD006517 was found in conode 53\n",
      "PXD006522 was found in conode 53\n",
      "PXD006552 was found in conode 53\n",
      "PXD006570 was found in conode 53\n",
      "PXD006571 was found in conode 53\n",
      "PXD006572 was found in conode 53\n",
      "PXD006576 was found in conode 53\n",
      "PXD006578 was found in conode 53\n",
      "PXD006581 was found in conode 54\n",
      "PXD006589 was found in conode 53\n",
      "PXD006607 was found in conode 54\n",
      "PXD006743 was found in conode 54\n",
      "PXD006776 was found in conode 53\n",
      "PXD006805 was found in conode 53\n",
      "PXD006833 was found in conode 53\n",
      "PXD006847 was found in conode 53\n",
      "PXD006897 was found in conode 53\n",
      "PXD006911 was found in conode 54\n",
      "PXD007080 was found in conode 54\n",
      "PXD007081 was found in conode 54\n",
      "PXD007122 was found in conode 54\n",
      "PXD007123 was found in conode 54\n",
      "PXD007137 was found in conode 53\n",
      "PXD007231 was found in conode 54\n",
      "PXD007234 was found in conode 54\n",
      "PXD007258 was found in conode 53\n",
      "PXD007515 was found in conode 54\n",
      "PXD007592 was found in conode 53\n",
      "PXD007705 was found in conode 54\n",
      "PXD007740 was found in conode 54\n",
      "PXD007744 was found in conode 53\n",
      "PXD007745 was found in conode 53\n",
      "PXD007846 was found in conode 53\n",
      "PXD007846 was found in conode 53\n",
      "PXD007873 was found in conode 54\n",
      "PXD007894 was found in conode 54\n",
      "PXD007935 was found in conode 53\n",
      "PXD007959 was found in conode 54\n",
      "PXD007975 was found in conode 54\n",
      "PXD008029 was found in conode 53\n",
      "PXD008029 was found in conode 53\n",
      "PXD008127 was found in conode 54\n",
      "PXD008130 was found in conode 54\n",
      "PXD008205 was found in conode 53\n",
      "PXD008252 was found in conode 54\n",
      "PXD008252 was found in conode 54\n",
      "PXD008311 was found in conode 53\n",
      "PXD008440 was found in conode 53\n",
      "PXD008443 was found in conode 54\n",
      "PXD008465 was found in conode 53\n",
      "PXD008467 was found in conode 54\n",
      "PXD008468 was found in conode 54\n",
      "PXD008493 was found in conode 53\n",
      "PXD008568 was found in conode 54\n",
      "PXD008581 was found in conode 53\n",
      "PXD008682 was found in conode 54\n",
      "PXD008723 was found in conode 53\n",
      "PXD008795 was found in conode 53\n",
      "PXD008824 was found in conode 54\n",
      "PXD008832 was found in conode 53\n",
      "PXD008934 was found in conode 53\n",
      "PXD008980 was found in conode 54\n",
      "PXD008992 was found in conode 54\n",
      "PXD008996 was found in conode 53\n",
      "PXD009021 was found in conode 53\n",
      "PXD009029 was found in conode 53\n",
      "PXD009056 was found in conode 54\n",
      "PXD009061 was found in conode 54\n",
      "PXD009112 was found in conode 54\n",
      "PXD009196 was found in conode 54\n",
      "PXD009196 was found in conode 54\n",
      "PXD009254 was found in conode 54\n",
      "PXD009258 was found in conode 54\n",
      "PXD009300 was found in conode 54\n",
      "PXD009310 was found in conode 53\n",
      "PXD009317 was found in conode 54\n",
      "PXD009343 was found in conode 53\n",
      "PXD009348 was found in conode 54\n",
      "PXD009350 was found in conode 53\n",
      "PXD009396 was found in conode 54\n",
      "PXD009438 was found in conode 54\n",
      "PXD009440 was found in conode 54\n",
      "PXD009444 was found in conode 54\n",
      "PXD009444 was found in conode 54\n",
      "PXD009476 was found in conode 54\n",
      "PXD009476 was found in conode 54\n",
      "PXD009606 was found in conode 54\n",
      "PXD009624 was found in conode 53\n",
      "PXD009624 was found in conode 53\n",
      "PXD009645 was found in conode 53\n",
      "PXD009645 was found in conode 53\n",
      "PXD009646 was found in conode 53\n",
      "PXD009709 was found in conode 54\n",
      "PXD009716 was found in conode 54\n",
      "PXD009737 was found in conode 53\n",
      "PXD009738 was found in conode 53\n",
      "PXD009738 was found in conode 53\n",
      "PXD009752 was found in conode 53\n",
      "PXD009754 was found in conode 53\n",
      "PXD009755 was found in conode 53\n",
      "PXD009784 was found in conode 54\n",
      "PXD009813 was found in conode 54\n",
      "PXD009819 was found in conode 54\n",
      "PXD009870 was found in conode 54\n",
      "PXD009880 was found in conode 54\n",
      "PXD009880 was found in conode 54\n",
      "PXD009893 was found in conode 54\n",
      "PXD009916 was found in conode 54\n",
      "PXD009917 was found in conode 54\n",
      "PXD009933 was found in conode 54\n",
      "PXD009996 was found in conode 53\n",
      "PXD010093 was found in conode 53\n",
      "PXD010142 was found in conode 53\n",
      "PXD010142 was found in conode 53\n",
      "PXD010150 was found in conode 53\n",
      "PXD010260 was found in conode 54\n",
      "PXD010260 was found in conode 54\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010271 was found in conode 53\n",
      "PXD010284 was found in conode 54\n",
      "PXD010286 was found in conode 54\n",
      "PXD010294 was found in conode 54\n",
      "PXD010469 was found in conode 54\n",
      "PXD010630 was found in conode 54\n",
      "PXD010690 was found in conode 53\n",
      "PXD010698 was found in conode 53\n",
      "PXD010705 was found in conode 54\n",
      "PXD010708 was found in conode 53\n",
      "PXD010786 was found in conode 53\n",
      "PXD010787 was found in conode 53\n",
      "PXD010821 was found in conode 54\n",
      "PXD011000 was found in conode 54\n",
      "PXD011152 was found in conode 54\n",
      "PXD011175 was found in conode 54\n",
      "PXD011183 was found in conode 53\n",
      "PXD011204 was found in conode 53\n",
      "PXD011214 was found in conode 54\n",
      "PXD011241 was found in conode 54\n",
      "PXD011309 was found in conode 54\n",
      "PXD011492 was found in conode 54\n",
      "PXD011609 was found in conode 54\n",
      "257\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    "m = (len(set(mapping_xl)))\n",
    "r = (len(set(reprocessed_projects)))\n",
    "print(m-r)\n",
    "print(r/m*100)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "288\n",
      "42.28456913827655\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "source": [
    "print(set(mapping_xl) - set(reprocessed_projects))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "{'PXD018445', 'PXD012029', 'PXD005698', 'PXD013840', 'PXD005949', 'PXD001630', 'PXD001183', 'PXD004818', 'PXD002614', 'PXD005973', 'PXD018021', 'PXD007982', 'PXD002255', 'PXD000437', 'PXD006286', 'PXD017030', 'PXD008700', 'PXD012574', 'PXD001323', 'PXD016278', 'PXD010086', 'PXD005520', 'PXD007232', 'PXD009378', 'PXD006353', 'PXD017201', 'PXD017805', 'PXD009203', 'PXD000755', 'PXD008570', 'PXD015578', 'PXD008036', 'PXD015780', 'PXD005613', 'PXD005274', 'PXD002798', 'PXD008820', 'PXD001034', 'PXD005013', 'PXD015289', 'PXD008278', 'PXD016864', 'PXD010252', 'PXD018357', 'PXD002372', 'PXD015049', 'PXD016158', 'PXD010480', 'PXD017268', 'PXD018588', 'PXD018369', 'PXD008146', 'PXD003218', 'PXD003014', 'PXD016698', 'PXD018226', 'PXD001124', 'PXD000383', 'PXD008821', 'PXD008891', 'PXD004465', 'PXD018208', 'PXD009303', 'PXD000742', 'PXD014524', 'PXD002421', 'PXD006083', 'PXD000701', 'PXD002911', 'PXD008625', 'PXD005692', 'PXD007839', 'PXD010965', 'PXD004312', 'PXD009707', 'PXD015880', 'PXD001536', 'PXD000551', 'PXD017195', 'PXD007134', 'PXD009074', 'PXD002884', 'PXD004369', 'PXD015024', 'PXD010437', 'PXD000603', 'PXD013038', 'PXD002781', 'PXD000963', 'PXD003340', 'PXD005978', 'PXD012971', 'PXD008141', 'PXD009382', 'PXD016261', 'PXD009896', 'PXD003651', 'PXD006482', 'PXD008822', 'PXD009681', 'PXD000457', 'PXD004998', 'PXD008407', 'PXD004987', 'PXD005086', 'PXD000226', 'PXD017217', 'PXD013649', 'PXD010626', 'PXD007223', 'PXD015653', 'PXD000395', 'PXD017023', 'PXD000743', 'PXD004935', 'PXD003924', 'PXD014300', 'PXD014894', 'PXD002163', 'PXD018502', 'PXD005777', 'PXD015836', 'PXD008023', 'PXD000117', 'PXD009723', 'PXD000656', 'PXD000746', 'PXD001311', 'PXD009602', 'PXD009345', 'PXD000654', 'PXD003785', 'PXD002367', 'PXD003519', 'PXD000655', 'PXD016999', 'PXD016731', 'PXD011116', 'PXD008383', 'PXD009969', 'PXD008773', 'PXD002424', 'PXD002523', 'PXD001539', 'PXD004205', 'PXD017710', 'PXD000809', 'PXD008788', 'PXD004219', 'PXD016865', 'PXD005426', 'PXD010230', 'PXD017799', 'PXD017149', 'PXD017051', 'PXD008485', 'PXD009051', 'PXD006757', 'PXD009227', 'PXD003271', 'PXD016859', 'PXD002236', 'PXD008293', 'PXD011178', 'PXD000653', 'PXD015239', 'PXD002346', 'PXD001402', 'PXD006687', 'PXD005073', 'PXD001435', 'PXD011187', 'PXD002171', 'PXD005847', 'PXD001997', 'PXD003552', 'PXD001099', 'PXD001282', 'PXD004815', 'PXD001353', 'PXD018678', 'PXD016217', 'PXD004684', 'PXD005150', 'PXD000657', 'PXD000749', 'PXD006367', 'PXD001474', 'PXD018682', 'PXD003613', 'PXD016435', 'PXD002312', 'PXD009598', 'PXD000757', 'PXD017968', 'PXD000302', 'PXD009971', 'PXD000314', 'PXD005267', 'PXD004249', 'PXD001468', 'PXD000190', 'PXD001271', 'PXD001499', 'PXD017213', 'PXD000534', 'PXD002533', 'PXD017804', 'PXD008450', 'PXD000807', 'PXD015163', 'PXD005992', 'PXD006471', 'PXD008720', 'PXD011129', 'PXD001694', 'PXD012461', 'PXD003121', 'PXD001449', 'PXD012960', 'PXD016697', 'PXD000768', 'PXD012970', 'PXD000651', 'PXD011749', 'PXD003224', 'PXD001506', 'PXD010737', 'PXD002633', 'PXD004397', 'PXD008512', 'PXD017269', 'PXD017976', 'PXD003269', 'PXD015523', 'PXD002673', 'PXD011228', 'PXD009450', 'PXD000307', 'PXD008553', 'PXD005970', 'PXD002629', 'PXD012615', 'PXD008022', 'PXD003029', 'PXD015518', 'PXD006333', 'PXD002050', 'PXD000921', 'PXD016314', 'PXD001052', 'PXD007083', 'PXD008823', 'PXD001464', 'PXD015213', 'PXD016229', 'PXD000535', 'PXD016437', 'PXD000744', 'PXD002339', 'PXD000765', 'PXD010417', 'PXD000652', 'PXD017656', 'PXD008513', 'PXD000263', 'PXD001328', 'PXD008310', 'PXD011126', 'PXD000536', 'PXD011755', 'PXD004092', 'PXD000769', 'PXD004873', 'PXD003419', 'PXD000930', 'PXD016406', 'PXD000760', 'PXD007048', 'PXD008076', 'PXD010164', 'PXD016126', 'PXD010450', 'PXD017056', 'PXD009261', 'PXD008445', 'PXD004785', 'PXD017011'}\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "source": [
    "import requests"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "source": [
    "x = requests.get('https://www.ebi.ac.uk/pride/ws/archive/project/list?query=PXD001034')\n",
    "x.text"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'{\"list\":[{\"accession\":\"PXD001034\",\"title\":\"Biomarkers for Cervical Cancer. iTRAQ Discovery.\",\"projectDescription\":\"We developed a discovery-validation mass-spectrometry based pipe\",\"publicationDate\":\"2014-10-06\",\"submissionType\":\"COMPLETE\",\"numAssays\":6,\"species\":[\"Homo sapiens (Human)\"],\"tissues\":[\"blood serum\"],\"ptmNames\":[\"Carbamidomethyl\",\"Phospho\",\"Acetyl\",\"Oxidation\"],\"instrumentNames\":[\"Agilent instrument model\"],\"projectTags\":[\"Biomedical\"]}]}'"
      ]
     },
     "metadata": {},
     "execution_count": 23
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Not all the PXDs from the metadata spreadsheet are actually reprocessed by ionbot, some do not contain the ionbot folder due to eg. not being HCD data. To counter this we first get the PXDs of all the reprocessed data and then get the metadata from the spreadsheet AND directly from PRIDE using the API"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "source": [
    "path53 = '/home/compomics/conode53_pride/PRIDE_DATA/'\n",
    "path54 = '/home/compomics/conode54_pride/PRIDE_DATA/'\n",
    "path55 = '/home/compomics/conode55_pride/PRIDE_DATA/'\n",
    "pxds53 = os.listdir(path53)\n",
    "pxds54 = os.listdir(path54)\n",
    "pxds55 = os.listdir(path55)\n",
    "allpxds = pxds53 + pxds54 + pxds55"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "source": [
    "type(pxds54)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "list"
      ]
     },
     "metadata": {},
     "execution_count": 58
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "source": [
    "count_all_reprocessed = 0\n",
    "all_reprocessed_projects = []\n",
    "for i in allpxds:\n",
    "    p = ''.join(list(i))\n",
    "    path53 = '/home/compomics/conode53_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path54 = '/home/compomics/conode54_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    path55 = '/home/compomics/conode55_pride/PRIDE_DATA/' + str(p) + '/IONBOT_v0.6.2/'\n",
    "    for file in glob.glob(path53):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            all_reprocessed_projects.append(p)\n",
    "            count_all_reprocessed += 1\n",
    "    for file in glob.glob(path54):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            all_reprocessed_projects.append(p)\n",
    "            count_all_reprocessed += 1\n",
    "    for file in glob.glob(path55):\n",
    "        if os.path.exists: #if this file exists in conode 53 and is thus reprocessed by ionbot\n",
    "            all_reprocessed_projects.append(p)\n",
    "            count_all_reprocessed += 1\n",
    "\n",
    "print(count_all_reprocessed)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "590\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "source": [
    "# get the information directly from PRIDE\n",
    "count = 0\n",
    "with open('pride_projects.txt', 'w') as file:\n",
    "    for i in all_reprocessed_projects:\n",
    "        x = requests.get('https://www.ebi.ac.uk/pride/ws/archive/project/list?query={}'.format(i))\n",
    "        count += 1\n",
    "        file.write((x.text[9:-2])+'\\n')\n",
    "        if count == 30:\n",
    "            time.sleep(61)\n",
    "            count = 0\n",
    "        print(count)\n"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "source": [
    "with open('pride_projects.txt', 'r') as f:\n",
    "    lines = f.read().split('\\n')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "source": [
    "import ast\n",
    "lines_dict = []\n",
    "for i in range(len(lines)):\n",
    "    print(i)\n",
    "    lines_dict.append(ast.literal_eval(lines[i]))"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n",
      "96\n",
      "97\n",
      "98\n",
      "99\n",
      "100\n",
      "101\n",
      "102\n",
      "103\n",
      "104\n",
      "105\n",
      "106\n",
      "107\n",
      "108\n",
      "109\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "116\n",
      "117\n",
      "118\n",
      "119\n",
      "120\n",
      "121\n",
      "122\n",
      "123\n",
      "124\n",
      "125\n",
      "126\n",
      "127\n",
      "128\n",
      "129\n",
      "130\n",
      "131\n",
      "132\n",
      "133\n",
      "134\n",
      "135\n",
      "136\n",
      "137\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "143\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "149\n",
      "150\n",
      "151\n",
      "152\n",
      "153\n",
      "154\n",
      "155\n",
      "156\n",
      "157\n",
      "158\n",
      "159\n",
      "160\n",
      "161\n",
      "162\n",
      "163\n",
      "164\n",
      "165\n",
      "166\n",
      "167\n",
      "168\n",
      "169\n",
      "170\n",
      "171\n",
      "172\n",
      "173\n",
      "174\n",
      "175\n",
      "176\n",
      "177\n",
      "178\n",
      "179\n",
      "180\n",
      "181\n",
      "182\n",
      "183\n",
      "184\n",
      "185\n",
      "186\n",
      "187\n",
      "188\n",
      "189\n",
      "190\n",
      "191\n",
      "192\n",
      "193\n",
      "194\n",
      "195\n",
      "196\n",
      "197\n",
      "198\n",
      "199\n",
      "200\n",
      "201\n",
      "202\n",
      "203\n",
      "204\n",
      "205\n",
      "206\n",
      "207\n",
      "208\n",
      "209\n",
      "210\n",
      "211\n",
      "212\n",
      "213\n",
      "214\n",
      "215\n",
      "216\n",
      "217\n",
      "218\n",
      "219\n",
      "220\n",
      "221\n",
      "222\n",
      "223\n",
      "224\n",
      "225\n",
      "226\n",
      "227\n",
      "228\n",
      "229\n",
      "230\n",
      "231\n",
      "232\n",
      "233\n",
      "234\n",
      "235\n",
      "236\n",
      "237\n",
      "238\n",
      "239\n",
      "240\n",
      "241\n",
      "242\n",
      "243\n",
      "244\n",
      "245\n",
      "246\n",
      "247\n",
      "248\n",
      "249\n",
      "250\n",
      "251\n",
      "252\n",
      "253\n",
      "254\n",
      "255\n",
      "256\n",
      "257\n",
      "258\n",
      "259\n",
      "260\n",
      "261\n",
      "262\n",
      "263\n",
      "264\n",
      "265\n",
      "266\n",
      "267\n",
      "268\n",
      "269\n",
      "270\n",
      "271\n",
      "272\n",
      "273\n",
      "274\n",
      "275\n",
      "276\n",
      "277\n",
      "278\n",
      "279\n",
      "280\n",
      "281\n",
      "282\n",
      "283\n",
      "284\n",
      "285\n",
      "286\n",
      "287\n",
      "288\n",
      "289\n",
      "290\n",
      "291\n",
      "292\n",
      "293\n",
      "294\n",
      "295\n",
      "296\n",
      "297\n",
      "298\n",
      "299\n",
      "300\n",
      "301\n",
      "302\n",
      "303\n",
      "304\n",
      "305\n",
      "306\n",
      "307\n",
      "308\n",
      "309\n",
      "310\n",
      "311\n",
      "312\n",
      "313\n",
      "314\n",
      "315\n",
      "316\n",
      "317\n",
      "318\n",
      "319\n",
      "320\n",
      "321\n",
      "322\n",
      "323\n",
      "324\n",
      "325\n",
      "326\n",
      "327\n",
      "328\n",
      "329\n",
      "330\n",
      "331\n",
      "332\n",
      "333\n",
      "334\n",
      "335\n",
      "336\n",
      "337\n",
      "338\n",
      "339\n",
      "340\n",
      "341\n",
      "342\n",
      "343\n",
      "344\n",
      "345\n",
      "346\n",
      "347\n",
      "348\n",
      "349\n",
      "350\n",
      "351\n",
      "352\n",
      "353\n",
      "354\n",
      "355\n",
      "356\n",
      "357\n",
      "358\n",
      "359\n",
      "360\n",
      "361\n",
      "362\n",
      "363\n",
      "364\n",
      "365\n",
      "366\n",
      "367\n",
      "368\n",
      "369\n",
      "370\n",
      "371\n",
      "372\n",
      "373\n",
      "374\n",
      "375\n",
      "376\n",
      "377\n",
      "378\n",
      "379\n",
      "380\n",
      "381\n",
      "382\n",
      "383\n",
      "384\n",
      "385\n",
      "386\n",
      "387\n",
      "388\n",
      "389\n",
      "390\n",
      "391\n",
      "392\n",
      "393\n",
      "394\n",
      "395\n",
      "396\n",
      "397\n",
      "398\n",
      "399\n",
      "400\n",
      "401\n",
      "402\n",
      "403\n",
      "404\n",
      "405\n",
      "406\n",
      "407\n",
      "408\n",
      "409\n",
      "410\n",
      "411\n",
      "412\n",
      "413\n",
      "414\n",
      "415\n",
      "416\n",
      "417\n",
      "418\n",
      "419\n",
      "420\n",
      "421\n",
      "422\n",
      "423\n",
      "424\n",
      "425\n",
      "426\n",
      "427\n",
      "428\n",
      "429\n",
      "430\n",
      "431\n",
      "432\n",
      "433\n",
      "434\n",
      "435\n",
      "436\n",
      "437\n",
      "438\n",
      "439\n",
      "440\n",
      "441\n",
      "442\n",
      "443\n",
      "444\n",
      "445\n",
      "446\n",
      "447\n",
      "448\n",
      "449\n",
      "450\n",
      "451\n",
      "452\n",
      "453\n",
      "454\n",
      "455\n",
      "456\n",
      "457\n",
      "458\n",
      "459\n",
      "460\n",
      "461\n",
      "462\n",
      "463\n",
      "464\n",
      "465\n",
      "466\n",
      "467\n",
      "468\n",
      "469\n",
      "470\n",
      "471\n",
      "472\n",
      "473\n",
      "474\n",
      "475\n",
      "476\n",
      "477\n",
      "478\n",
      "479\n",
      "480\n",
      "481\n",
      "482\n",
      "483\n",
      "484\n",
      "485\n",
      "486\n",
      "487\n",
      "488\n",
      "489\n",
      "490\n",
      "491\n",
      "492\n",
      "493\n",
      "494\n",
      "495\n",
      "496\n",
      "497\n",
      "498\n",
      "499\n",
      "500\n",
      "501\n",
      "502\n",
      "503\n",
      "504\n",
      "505\n",
      "506\n",
      "507\n",
      "508\n",
      "509\n",
      "510\n",
      "511\n",
      "512\n",
      "513\n",
      "514\n",
      "515\n",
      "516\n",
      "517\n",
      "518\n",
      "519\n",
      "520\n",
      "521\n",
      "522\n",
      "523\n",
      "524\n",
      "525\n",
      "526\n",
      "527\n",
      "528\n",
      "529\n",
      "530\n",
      "531\n",
      "532\n",
      "533\n",
      "534\n",
      "535\n",
      "536\n",
      "537\n",
      "538\n",
      "539\n",
      "540\n",
      "541\n",
      "542\n",
      "543\n",
      "544\n",
      "545\n",
      "546\n",
      "547\n",
      "548\n",
      "549\n",
      "550\n",
      "551\n",
      "552\n",
      "553\n",
      "554\n",
      "555\n",
      "556\n",
      "557\n",
      "558\n",
      "559\n",
      "560\n",
      "561\n",
      "562\n",
      "563\n",
      "564\n",
      "565\n",
      "566\n",
      "567\n",
      "568\n",
      "569\n",
      "570\n",
      "571\n",
      "572\n",
      "573\n",
      "574\n",
      "575\n",
      "576\n",
      "577\n",
      "578\n",
      "579\n",
      "580\n",
      "581\n",
      "582\n",
      "583\n",
      "584\n",
      "585\n",
      "586\n",
      "587\n",
      "588\n",
      "589\n",
      "590\n",
      "591\n",
      "592\n",
      "593\n"
     ]
    },
    {
     "output_type": "error",
     "ename": "SyntaxError",
     "evalue": "unexpected EOF while parsing (<unknown>, line 0)",
     "traceback": [
      "Traceback \u001b[0;36m(most recent call last)\u001b[0m:\n",
      "  File \u001b[1;32m\"/home/compomics/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py\"\u001b[0m, line \u001b[1;32m3418\u001b[0m, in \u001b[1;35mrun_code\u001b[0m\n    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \u001b[1;32m\"<ipython-input-17-e800b0559c4a>\"\u001b[0m, line \u001b[1;32m5\u001b[0m, in \u001b[1;35m<module>\u001b[0m\n    lines_dict.append(ast.literal_eval(lines[i]))\n",
      "  File \u001b[1;32m\"/home/compomics/miniconda3/lib/python3.7/ast.py\"\u001b[0m, line \u001b[1;32m46\u001b[0m, in \u001b[1;35mliteral_eval\u001b[0m\n    node_or_string = parse(node_or_string, mode='eval')\n",
      "\u001b[0;36m  File \u001b[0;32m\"/home/compomics/miniconda3/lib/python3.7/ast.py\"\u001b[0;36m, line \u001b[0;32m35\u001b[0;36m, in \u001b[0;35mparse\u001b[0;36m\u001b[0m\n\u001b[0;31m    return compile(source, filename, mode, PyCF_ONLY_AST)\u001b[0m\n",
      "\u001b[0;36m  File \u001b[0;32m\"<unknown>\"\u001b[0;36m, line \u001b[0;32munknown\u001b[0m\n\u001b[0;31m    \u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m unexpected EOF while parsing\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "source": [
    "for i in lines_dict:\n",
    "    if isinstance(i, tuple):\n",
    "        print(i)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "source": [
    "df = pd.DataFrame(lines_dict)\n",
    "df = df.loc[df.astype(str).drop_duplicates().index]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "source": [
    "#only retain PXDs with tissue annotation\n",
    "df_annotated = df[~df.tissues.str.len().eq(0)]\n",
    "#only retain pxds with ONLY human proteins\n",
    "df_annotated['species'] = df_annotated['species'].astype('str')\n",
    "df_annotated = df_annotated.loc[df_annotated['species'] == \"['Homo sapiens (Human)']\"]"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "/home/compomics/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "source": [
    "df_annotated.sort_values('tissues')"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>title</th>\n",
       "      <th>projectDescription</th>\n",
       "      <th>publicationDate</th>\n",
       "      <th>submissionType</th>\n",
       "      <th>numAssays</th>\n",
       "      <th>species</th>\n",
       "      <th>tissues</th>\n",
       "      <th>ptmNames</th>\n",
       "      <th>instrumentNames</th>\n",
       "      <th>projectTags</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>573</th>\n",
       "      <td>PXD006269</td>\n",
       "      <td>HCMV pentamer - Identification of protein inte...</td>\n",
       "      <td>Identification of protein interacting with gHg...</td>\n",
       "      <td>2018-07-27</td>\n",
       "      <td>PARTIAL</td>\n",
       "      <td>0</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[ARPE-19 cell]</td>\n",
       "      <td>[amidated residue, acetylated residue]</td>\n",
       "      <td>[Q Exactive]</td>\n",
       "      <td>[Biological]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PXD005389</td>\n",
       "      <td>Targeting the Scaffold Protein Plectin by an O...</td>\n",
       "      <td>The inability of determining direct molecular ...</td>\n",
       "      <td>2017-06-02</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>4</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[HCT-116 cell, colon]</td>\n",
       "      <td>[Carbamidomethyl, Acetyl, Oxidation]</td>\n",
       "      <td>[Q Exactive]</td>\n",
       "      <td>[Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PXD004764</td>\n",
       "      <td>Response Profiling Using Shotgun Proteomics En...</td>\n",
       "      <td>Response profiling using shotgun proteomics fo...</td>\n",
       "      <td>2017-01-11</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>6</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[HCT-116 cell, colon]</td>\n",
       "      <td>[Carbamidomethyl, Acetyl, Oxidation]</td>\n",
       "      <td>[Q Exactive]</td>\n",
       "      <td>[Biological, Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PXD006299</td>\n",
       "      <td>Characterisation of protein ubiquitination usi...</td>\n",
       "      <td>We present a novel immune-affinity-based metho...</td>\n",
       "      <td>2018-07-03</td>\n",
       "      <td>PARTIAL</td>\n",
       "      <td>0</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[HEp-2 cell, JURKAT cell]</td>\n",
       "      <td>[ubiquitination signature dipeptidyl lysine, p...</td>\n",
       "      <td>[Q Exactive HF]</td>\n",
       "      <td>[Technical, Biological]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>PXD001249</td>\n",
       "      <td>SNW1 is required for sister chromatid cohesion</td>\n",
       "      <td>Although splicing is essential for the express...</td>\n",
       "      <td>2016-06-29</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>2</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[HeLa cell]</td>\n",
       "      <td>[deaminated residue, monohydroxylated residue,...</td>\n",
       "      <td>[Q Exactive]</td>\n",
       "      <td>[Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>PXD011214</td>\n",
       "      <td>Urinary glycopeptide analysis for the investig...</td>\n",
       "      <td>The aim of this study was to investigate the p...</td>\n",
       "      <td>2018-11-29</td>\n",
       "      <td>PARTIAL</td>\n",
       "      <td>0</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[urine]</td>\n",
       "      <td>[Hex1HexNAc2, monohydroxylated residue, Hex1He...</td>\n",
       "      <td>[LTQ Orbitrap Velos]</td>\n",
       "      <td>[Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>PXD003045</td>\n",
       "      <td>SPE optimization for peptide sample preparatio...</td>\n",
       "      <td>Development of a modified solid phase extracti...</td>\n",
       "      <td>2015-10-16</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>8</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[urine]</td>\n",
       "      <td>[Oxidation]</td>\n",
       "      <td>[LTQ Orbitrap Velos]</td>\n",
       "      <td>[Technical, Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>PXD009709</td>\n",
       "      <td>Ultrafiltration and size exclusion chromatogra...</td>\n",
       "      <td>Extracellular vesicles (EVs) have a great pote...</td>\n",
       "      <td>2018-07-04</td>\n",
       "      <td>PARTIAL</td>\n",
       "      <td>0</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[urine]</td>\n",
       "      <td>[iodoacetamide derivatized residue, phosphoryl...</td>\n",
       "      <td>[Q Exactive]</td>\n",
       "      <td>[Technical, Biological]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>PXD008723</td>\n",
       "      <td>Proteomic alterations in early stage cervical ...</td>\n",
       "      <td>Cervical cancer is characterized by a well-def...</td>\n",
       "      <td>2018-05-02</td>\n",
       "      <td>COMPLETE</td>\n",
       "      <td>36</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[uterine cervix]</td>\n",
       "      <td>[Oxidation]</td>\n",
       "      <td>[Orbitrap Fusion]</td>\n",
       "      <td>[Biological, Biomedical]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>PXD008354</td>\n",
       "      <td>The Proteomic Landscape in the Vitreous of Pat...</td>\n",
       "      <td>In contrast to neovascular age-related macular...</td>\n",
       "      <td>2018-07-23</td>\n",
       "      <td>PARTIAL</td>\n",
       "      <td>0</td>\n",
       "      <td>['Homo sapiens (Human)']</td>\n",
       "      <td>[vitreous humor]</td>\n",
       "      <td>[iodoacetamide derivatized residue, monohydrox...</td>\n",
       "      <td>[Orbitrap Fusion]</td>\n",
       "      <td>[Biomedical]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>410 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     accession                                              title  \\\n",
       "573  PXD006269  HCMV pentamer - Identification of protein inte...   \n",
       "2    PXD005389  Targeting the Scaffold Protein Plectin by an O...   \n",
       "34   PXD004764  Response Profiling Using Shotgun Proteomics En...   \n",
       "13   PXD006299  Characterisation of protein ubiquitination usi...   \n",
       "115  PXD001249     SNW1 is required for sister chromatid cohesion   \n",
       "..         ...                                                ...   \n",
       "585  PXD011214  Urinary glycopeptide analysis for the investig...   \n",
       "65   PXD003045  SPE optimization for peptide sample preparatio...   \n",
       "355  PXD009709  Ultrafiltration and size exclusion chromatogra...   \n",
       "171  PXD008723  Proteomic alterations in early stage cervical ...   \n",
       "534  PXD008354  The Proteomic Landscape in the Vitreous of Pat...   \n",
       "\n",
       "                                    projectDescription publicationDate  \\\n",
       "573  Identification of protein interacting with gHg...      2018-07-27   \n",
       "2    The inability of determining direct molecular ...      2017-06-02   \n",
       "34   Response profiling using shotgun proteomics fo...      2017-01-11   \n",
       "13   We present a novel immune-affinity-based metho...      2018-07-03   \n",
       "115  Although splicing is essential for the express...      2016-06-29   \n",
       "..                                                 ...             ...   \n",
       "585  The aim of this study was to investigate the p...      2018-11-29   \n",
       "65   Development of a modified solid phase extracti...      2015-10-16   \n",
       "355  Extracellular vesicles (EVs) have a great pote...      2018-07-04   \n",
       "171  Cervical cancer is characterized by a well-def...      2018-05-02   \n",
       "534  In contrast to neovascular age-related macular...      2018-07-23   \n",
       "\n",
       "    submissionType  numAssays                   species  \\\n",
       "573        PARTIAL          0  ['Homo sapiens (Human)']   \n",
       "2         COMPLETE          4  ['Homo sapiens (Human)']   \n",
       "34        COMPLETE          6  ['Homo sapiens (Human)']   \n",
       "13         PARTIAL          0  ['Homo sapiens (Human)']   \n",
       "115       COMPLETE          2  ['Homo sapiens (Human)']   \n",
       "..             ...        ...                       ...   \n",
       "585        PARTIAL          0  ['Homo sapiens (Human)']   \n",
       "65        COMPLETE          8  ['Homo sapiens (Human)']   \n",
       "355        PARTIAL          0  ['Homo sapiens (Human)']   \n",
       "171       COMPLETE         36  ['Homo sapiens (Human)']   \n",
       "534        PARTIAL          0  ['Homo sapiens (Human)']   \n",
       "\n",
       "                       tissues  \\\n",
       "573             [ARPE-19 cell]   \n",
       "2        [HCT-116 cell, colon]   \n",
       "34       [HCT-116 cell, colon]   \n",
       "13   [HEp-2 cell, JURKAT cell]   \n",
       "115                [HeLa cell]   \n",
       "..                         ...   \n",
       "585                    [urine]   \n",
       "65                     [urine]   \n",
       "355                    [urine]   \n",
       "171           [uterine cervix]   \n",
       "534           [vitreous humor]   \n",
       "\n",
       "                                              ptmNames       instrumentNames  \\\n",
       "573             [amidated residue, acetylated residue]          [Q Exactive]   \n",
       "2                 [Carbamidomethyl, Acetyl, Oxidation]          [Q Exactive]   \n",
       "34                [Carbamidomethyl, Acetyl, Oxidation]          [Q Exactive]   \n",
       "13   [ubiquitination signature dipeptidyl lysine, p...       [Q Exactive HF]   \n",
       "115  [deaminated residue, monohydroxylated residue,...          [Q Exactive]   \n",
       "..                                                 ...                   ...   \n",
       "585  [Hex1HexNAc2, monohydroxylated residue, Hex1He...  [LTQ Orbitrap Velos]   \n",
       "65                                         [Oxidation]  [LTQ Orbitrap Velos]   \n",
       "355  [iodoacetamide derivatized residue, phosphoryl...          [Q Exactive]   \n",
       "171                                        [Oxidation]     [Orbitrap Fusion]   \n",
       "534  [iodoacetamide derivatized residue, monohydrox...     [Orbitrap Fusion]   \n",
       "\n",
       "                  projectTags  \n",
       "573              [Biological]  \n",
       "2                [Biomedical]  \n",
       "34   [Biological, Biomedical]  \n",
       "13    [Technical, Biological]  \n",
       "115              [Biomedical]  \n",
       "..                        ...  \n",
       "585              [Biomedical]  \n",
       "65    [Technical, Biomedical]  \n",
       "355   [Technical, Biological]  \n",
       "171  [Biological, Biomedical]  \n",
       "534              [Biomedical]  \n",
       "\n",
       "[410 rows x 11 columns]"
      ]
     },
     "metadata": {},
     "execution_count": 61
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "source": [
    "df_annotated.to_csv('direct_from_pride_metadata.csv',sep='\\t', index=False)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Now we have all the metadata from the reprocessed projects, lets also extract the metadata from the metdata spreadsheet"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "source": [
    "meta_ionbot = meta\n",
    "meta_pride = pd.read_csv('direct_from_pride_metadata.csv', sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "source": [
    "print(meta_ionbot.columns)\n",
    "print(meta_pride.columns)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "Index(['accession', 'url', 'title', 'projectDescription', 'species', 'tissues',\n",
      "       'ptms', 'instrumentNames', 'projectTags', 'experimentTypes',\n",
      "       'quantificationMethods', 'keywords', 'dataProcessingProtocol',\n",
      "       'submissionDate', 'references'],\n",
      "      dtype='object')\n",
      "Index(['accession', 'title', 'projectDescription', 'publicationDate',\n",
      "       'submissionType', 'numAssays', 'species', 'tissues', 'ptmNames',\n",
      "       'instrumentNames', 'projectTags'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "source": [
    "meta_complete = meta_pride[['accession', 'species', 'tissues', 'ptmNames']].merge(meta_ionbot, on='accession', how='left')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "source": [
    "print(meta_ionbot.shape, meta_pride.shape, meta_complete.shape) #check if all the accessions were found\n",
    "c = set(list(meta_complete['accession'].values))\n",
    "p = set(list(meta_pride['accession'].values))\n",
    "print(len(c), len(p))\n",
    "print(c-p)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(2253, 15) (410, 11) (411, 18)\n",
      "410 410\n",
      "set()\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Now we should also add the textmining results"
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "projects were selected based on tissue = 1 and diseased = 0 <br>\n",
    "85 projects were found, now check if they are in the reprocessed output"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "source": [
    "text_projects = ['PXD011755','PXD011467','PXD011456','PXD011309','PXD011126','PXD011126','PXD011118','PXD010870','PXD010516','PXD009969','PXD009880','PXD009870','PXD009868','PXD009738','PXD009724','PXD009707','PXD009707','PXD009603','PXD009603','PXD009396','PXD009396','PXD009396','PXD009350','PXD009251','PXD009203','PXD009196','PXD009172','PXD009024','PXD008980','PXD008821','PXD008788','PXD008723','PXD008720','PXD008720','PXD008581','PXD008513','PXD008513','PXD008014','PXD007234','PXD007134','PXD007130','PXD006863','PXD006769','PXD006673','PXD006578','PXD006578','PXD005755','PXD005755','PXD005613','PXD005267','PXD005013','PXD004874','PXD004626','PXD004588','PXD004443','PXD004312','PXD004219','PXD003534','PXD003340','PXD003218','PXD002781','PXD002672','PXD002633','PXD002437','PXD002163','PXD002014','PXD002014','PXD001674','PXD001524','PXD001425','PXD001362','PXD001357','PXD000688','PXD000656','PXD000654','PXD000653','PXD000653','PXD000651','PXD000651','PXD000603','PXD000534','PXD000385','PXD000228','PXD000226','PXD000072']\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "source": [
    "textmining_projects = []\n",
    "for i in text_projects:\n",
    "    if i in all_reprocessed_projects:\n",
    "        textmining_projects.append(i)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "source": [
    "len(textmining_projects)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "41"
      ]
     },
     "metadata": {},
     "execution_count": 96
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "source": [
    "text = pd.read_csv('full_data_textmining.csv', sep='/', encoding = \"ISO-8859-1\")\n",
    "text = text.rename(columns={'pride_accession': 'accession'})"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "source": [
    "meta_text = text[text['accession'].isin(textmining_projects)] #textmining of reprocessed projects\n",
    "x = meta_complete['accession'].tolist() #projects in \n",
    "meta_text_2 = text[text['accession'].isin(x)] "
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "source": [
    "meta_complete_2 = meta_complete.merge(meta_text, on='accession', how='left')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "source": [
    "print(meta_complete.shape, meta_text.shape, meta_complete_2.shape)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(411, 18) (38, 36) (414, 53)\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "source": [
    "meta_complete_2.to_csv('metadata.csv', index=False, sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Earlier anual mapping of several projects to the HPA tissue led to the file PXD_HPA_tissue_mapping.csv . This can be used to find easily to bigger tissue to the current found projects in meta_complete_2 <br>\n",
    "Remove all the tissues that only say cell culture"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "source": [
    "map = pd.read_csv('PXD_HPA_tissue_mapping.csv',sep='\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "source": [
    "meta_complete_2.drop(meta_complete_2.loc[meta_complete_2['tissues_y']== '[\"cell culture\"]'].index, inplace=True)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "source": [
    "meta_complete_3 = pd.merge(meta_complete_2, map, on='accession', how='left')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "source": [
    "meta_complete_3.isna().sum()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "accession                                 0\n",
       "species_x                                 0\n",
       "tissues_x                                 0\n",
       "ptmNames                                  0\n",
       "url                                       4\n",
       "title                                     4\n",
       "projectDescription                        4\n",
       "species_y                                 3\n",
       "tissues_y                                 4\n",
       "ptms                                     27\n",
       "instrumentNames                           4\n",
       "projectTags                              55\n",
       "experimentTypes                           4\n",
       "quantificationMethods                   199\n",
       "keywords_x                                4\n",
       "dataProcessingProtocol                    4\n",
       "submissionDate                            4\n",
       "references                               25\n",
       "pubmed_id                               291\n",
       "article_name                            291\n",
       "species_name                            291\n",
       "species_ontology                        291\n",
       "tissue_cell_line                        291\n",
       "tissue_ontology                         291\n",
       "cell_line_ontology                      323\n",
       "labelling_name                          309\n",
       "labelling_ontology                      309\n",
       "enzyme_name                             298\n",
       "enzyme_ontology                         299\n",
       "no_missed_cleavages                     302\n",
       "precursor_mass_tolerance_ppm            299\n",
       "fragment_ion_mass_tolerance_da          301\n",
       "precursor_mass_tolerance_original       299\n",
       "fragment_ion_mass_tolerance_original    301\n",
       "variable_modification_name              298\n",
       "variable_modification_ontology          298\n",
       "fixed_modification_name                 298\n",
       "fixed_modification_ontology             298\n",
       "experiment_name                         304\n",
       "experiment_ontology                     304\n",
       "instrument_name                         294\n",
       "instrument_ontology                     294\n",
       "fragmentation_type_name                 296\n",
       "description                             291\n",
       "keywords_y                              291\n",
       "comment                                 307\n",
       "multiple_projects                       291\n",
       "open_access                             291\n",
       "pride_search_settings                   291\n",
       "cross_link                              291\n",
       "retracted_paper                         291\n",
       "synthetic_peptide_used.                 291\n",
       "student                                 291\n",
       "HPA tissue                               71\n",
       "dtype: int64"
      ]
     },
     "metadata": {},
     "execution_count": 149
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "source": [
    "meta_complete_3.to_csv('meta_complete_3.csv', sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "The metafile contains several projects with their tissue annotation. (not all HPA tissues were found at the moment) <br>\n",
    "We need to find a method of raw file tissue annotation."
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "\n"
   ],
   "outputs": [],
   "metadata": {}
  }
 ]
}